---
title: "new_dge_cglabrata"
author: "Weronika"
date: "2024-05-23"
output: 
  html_document:
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
options(knitr.duplicate.label = "allow")
knitr::opts_chunk$set(echo = TRUE)

library(tidyr)
library(tidyverse)
library(cowplot)
library(dplyr)
library(DESeq2)
library(GGally)
library(here)
library(biobroom)

theme_set(theme_cowplot(font_size = 12) +
            theme(strip.background = element_blank(),
                  axis.text=element_text(size=10),
                  axis.title=element_text(size=10),
                  plot.title = element_text(size=10),
                  legend.title=element_text(size=10),
                  legend.text=element_text(size=10)))

COUNT_HEADER_COLUMNS <- c("Geneid", "Chr", "Start",
                          "End", "Strand", "Length")

#knitr::knit_child("featureCounts_preprocess.Rmd")
#knitr::knit_child("ortho_preprocess.Rmd")
```

# Introduction

Candida glabrata strains:

1. CBS138
2. BG2

Treated with voriconazole (VORI), fluconazole (FLUC) or DMSO (solvent control).

Sequencing FASTQ files were aligned to appropriate genomes using HISAT2 and quantified using featureCounts.

MultiQC results, including quality control of raw reads (FastQC) and alignment are here:

- CBS138: `r paste('..','data', 'multiQC','Candida_glabrata_CBS138','multiqc_report.html', sep="/")`)
- BG2: `r paste('..','data', 'multiQC','Candida_glabrata_BG2','multiqc_report.html', sep="/")`)

featureCount results, aggregated by strain, are below:

```{r show_data_1}
counts_all_1 = read.table("../data/featureCounts/Candida_glabrata_CBS138/CBS138_all_conditions_featureCounts.tab", header=TRUE)
counts_all_2 = read.table("../data/featureCounts/Candida_glabrata_BG2/BG2_all_conditions_featureCounts.tab", header=TRUE)

counts_rpm_1 = read.table("../data/featureCounts/Candida_glabrata_CBS138/CBS138_all_conditions_rpm.tab", header=TRUE, row.names = 1)
counts_rpm_2 = read.table("../data/featureCounts/Candida_glabrata_BG2/BG2_all_conditions_rpm.tab", header=TRUE, row.names = 1)

head(counts_all_1) # CBS138, raw
head(counts_all_2) # BG2, raw
```

Results normalised to RPM are below:

```{r show_data}

head(counts_rpm_1) # CBS138, normalised to RPM
head(counts_rpm_2) # BG2, normalised to RPM

```

# Analysis

## Quality control - sample correlation

For quality control, we check correlation between the samples (only within strain).

There is high correlation within untreated (DMSO) samples and treated (FLUC and VORI), but no obvious differences between VORI and FLUC.


```{python pairplot1, fig.height = 4, fig.width=10, warning=FALSE, message=FALSE, echo=FALSE, eval=FALSE}
# TODO: no Python code in this file; eval set to FALSE until ported to R
from matplotlib import pyplot as plt
import seaborn as sns

plt.close()
#sns.pairplot(counts_only, corner=True, kind="scatter")

fig, ax = plt.subplots(1, 2, figsize=(9, 4), layout="tight")

heatmap = sns.heatmap(counts_1.corr(), annot=True, ax=ax[0], vmin=0.7, vmax=1.0);
heatmap.set_title('CBS138 Correlation Heatmap', fontdict={'fontsize':12}, pad=12);

heatmap = sns.heatmap(counts_2.corr(), annot=True, ax=ax[1], vmin=0.7, vmax=1.0);
heatmap.set_title('BG2 Correlation Heatmap', fontdict={'fontsize':12}, pad=12);

labs = ['DMSO_A', 'VORI_A', 'FLUC_A', 'DMSO_B', 'VORI_B', 'FLUC_B', 'DMSO_C', 'VORI_C', 'FLUC_C']
ax[0].set_xticks(ax[0].get_xticks(), labs, rotation=30, rotation_mode='anchor', ha='right')
ax[0].set_yticks(ax[0].get_yticks(), labs, rotation=30, rotation_mode='anchor', ha='right')
ax[1].set_xticks(ax[0].get_xticks(), labs, rotation=30, rotation_mode='anchor', ha='right')
ax[1].set_yticks(ax[0].get_yticks(), labs, rotation=30, rotation_mode='anchor', ha='right')
ax[0].tick_params(labelsize=8)
ax[1].tick_params(labelsize=8)

plt.show()
plt.close()
```



(This is a figure in progress)

When we cluster the results, again untreated samples cluster together and treated samples cluster together.
Treated samples cluster in a way that doesn't indicate any obvious sample mix-up.

CBS138_RPMI_VORI_C is an outlier.


```{r heatmap, fig.height = 10, fig.width=10, echo=FALSE, message=FALSE}
library(pheatmap)


# create heatmap using pheatmap
ph1 = pheatmap(counts_rpm_1, main="CBS138")
ph2 = pheatmap(counts_rpm_2, main="BG2")
```



## Differential Gene Expression analysis using DESeq2 in each strain separately

Differential Gene Expression analysis was performed using DESeq2.
Genes with adjusted p-value < 0.05 and at least 2x change are considered differentially expressed here.
The upregulated and downregulated genes are saved in ../data/deseq2


```{r functions, echo = FALSE}
# wrapper functions for DESeq2 package functions


de_seq <- function(counts_all, conditions_to_compare) {
  sample_sheet_compared_conditions <-
    dplyr::filter(samplesheet,
                  Condition %in% conditions_to_compare)
  rownames(sample_sheet_compared_conditions) <- 1:nrow(sample_sheet_compared_conditions)

  dds_compared_conditions <- 
    DESeqDataSetFromMatrix(countData = counts_all %>%
                             dplyr::select(sample_sheet_compared_conditions$Code) %>%
                             magrittr::set_rownames(counts_all$Gene),
                           colData = sample_sheet_compared_conditions,
                           design = ~ Medium) %>%
    DESeq()
  dds_compared_conditions
}

deseq_df_transform <- function(dds_compared_conditions) {
  deseq_df <- 
  biobroom::tidy.DESeqDataSet(dds_compared_conditions) %>%
  dplyr::filter(term == rN[2]) %>%
  dplyr::select(gene, 
                baseMean, 
                log2FC = estimate, 
                stderror,
                padj = p.adjusted)
  deseq_df
}

get_upregulated_genes <- function(deseq_df) {
  DEGdf_up2x_FDR5 <- 
    deseq_df %>%
    dplyr::filter(log2FC > 1, padj < 0.05) %>%
    dplyr::arrange(desc(log2FC))
  DEGdf_up2x_FDR5
}

get_downregulated_genes <-function(deseq_df) {
  DEGdf_down2x_FDR5 <- 
    deseq_df %>%
    dplyr::filter(log2FC < -1, padj < 0.05) %>%
    dplyr::arrange(log2FC)
    
  DEGdf_down2x_FDR5
}
  
get_non_significant_genes <- function(deseq_df) {
  DEGdf_non_significant <- 
    deseq_df %>%
    dplyr::filter(log2FC > -1, , log2FC < 1, padj > 0.05) %>%
    dplyr::arrange(log2FC)
  DEGdf_non_significant
}

volcano_plot <- function(deseq_df, rN2, up, down) {
  volcano_plot <- 
    ggplot(data = deseq_df,
           aes(x = log2FC, y = -log10(padj)) ) +
    geom_point(size = 0.5, colour = "grey50") +
    geom_point(data = up,
               size = 1, colour = "darkblue") +
    geom_point(data = down,
               size = 1, colour = "darkred") +
    scale_y_continuous("-log10(p)",
                       limits = c(0,50), expand = c(0,0),
                       oob=scales::squish) +
    scale_x_continuous("log2 fold-change",
                       breaks = -5:5,
                       limits = c(-4,4), expand = c(0.01,0.01),
                       oob = scales::squish) +
    coord_cartesian(clip = "off") 
  volcano_plot
  }


```


```{r cbs_f_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

samplesheet = read.csv("../data/sample_sheets/experiment_data_1.csv")
# samplesheet = py$experiment_data_1

counts_all = counts_all_1
head(counts_all)

conditions_to_compare = c("CBS138_RPMI_DMSO", "CBS138_RPMI_FLUC")


dds_compared_conditions = de_seq(counts_all, conditions_to_compare)
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_cf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/Candida_glabrata_CBS138/CBS138_FLUC_DMSO_allDESeq.csv")


p1 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_cf, DEGdf_down2x_FDR5_cf) +
  ggtitle(paste0(unlist(strsplit(rN[2], "_")), collapse = " "))
#p1
```

```{r cbs_v_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

samplesheet = read.csv("../data/sample_sheets/experiment_data_1.csv")
# samplesheet = py$experiment_data_1
counts_all = counts_all_1
head(counts_all)

conditions_to_compare = c("CBS138_RPMI_DMSO", "CBS138_RPMI_VORI")


dds_compared_conditions = de_seq(counts_all, conditions_to_compare)
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_cv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cv = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/Candida_glabrata_CBS138/CBS138_VORI_DMSO_allDESeq.csv")


p2 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_cv, DEGdf_down2x_FDR5_cv) +
  ggtitle(paste0(unlist(strsplit(rN[2], "_")), collapse = " "))
#p2

```



```{r cbs_vf_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

samplesheet = read.csv("../data/sample_sheets/experiment_data_1.csv")
# samplesheet = py$experiment_data_1
counts_all = counts_all_1
head(counts_all)

conditions_to_compare = c("CBS138_RPMI_FLUC", "CBS138_RPMI_VORI")


dds_compared_conditions = de_seq(counts_all, conditions_to_compare)
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_cvf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cvf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cvf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/Candida_glabrata_CBS138/CBS138_VORI_FLUC_allDESeq.csv")

p3 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_cvf, DEGdf_down2x_FDR5_cvf) +
  ggtitle(paste0(unlist(strsplit(rN[2], "_")), collapse = " "))
#p3

```

```{r bg2_f_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

samplesheet = read.csv("../data/sample_sheets/experiment_data_2.csv")
# samplesheet = py$experiment_data_2
counts_all = counts_all_2
head(counts_all)

conditions_to_compare = c("BG2_RPMI_DMSO", "BG2_RPMI_FLUC")


dds_compared_conditions = de_seq(counts_all, conditions_to_compare)
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_bf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_bf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_bf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/Candida_glabrata_CBS138/BG2_FLUC_DMSO_allDESeq.csv")

p4 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_bf, DEGdf_down2x_FDR5_bf) +
  ggtitle(paste0(unlist(strsplit(rN[2], "_")), collapse = " "))
#p4
```

```{r bg2_v_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

samplesheet = read.csv("../data/sample_sheets/experiment_data_2.csv")
# samplesheet = py$experiment_data_2
counts_all = counts_all_2
head(counts_all)

conditions_to_compare = c("BG2_RPMI_DMSO", "BG2_RPMI_VORI")


dds_compared_conditions = de_seq(counts_all, conditions_to_compare)
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_bv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_bv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_bv = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/Candida_glabrata_CBS138/BG2_VORI_DMSO_allDESeq.csv")

p5 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_bv, DEGdf_down2x_FDR5_bv) +
  ggtitle(paste0(unlist(strsplit(rN[2], "_")), collapse = " "))
#p5
```

```{r bg2_fv_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

samplesheet = read.csv("../data/sample_sheets/experiment_data_2.csv")
# samplesheet = py$experiment_data_2
counts_all = counts_all_2
head(counts_all)

conditions_to_compare = c("BG2_RPMI_FLUC", "BG2_RPMI_VORI")


dds_compared_conditions = de_seq(counts_all, conditions_to_compare)
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_bfv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_bfv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_bfv = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/Candida_glabrata_CBS138/BG2_VORI_FLUC_allDESeq.csv")

p6 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_bfv, DEGdf_down2x_FDR5_bfv) +
  ggtitle(paste0(unlist(strsplit(rN[2], "_")), collapse = " "))
# p6
```

```{r arrange_plots_1, fig.height=9, fig.width=8, echo=FALSE, warning=FALSE}
pg = plot_grid(p1, p4, p2, p5, p3, p6, ncol = 2, nrow = 3, rel_widths = c(5, 5))
pg
```

How many genes were differentially expressed in each condition? [TODO: make it a figure]

CBS138:

```{r cbs_f_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated CBS138 FLUC vs DMSO: %d", nrow(DEGdf_up2x_FDR5_cf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_cf, n = 10)


sprintf("Downregulated CBS138 FLUC vs DMSO: %d", nrow(DEGdf_down2x_FDR5_cf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_cf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_cf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_cf, "../data/deseq2/Candida_glabrata_CBS138/CBS138_FLUC_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_cf, "../data/deseq2/Candida_glabrata_CBS138/CBS138_FLUC_DMSO_down.csv")

```

```{r cbs_v_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in CBS138 VORI vs DMSO: %d", nrow(DEGdf_up2x_FDR5_cv))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_cv, n = 10)


sprintf("Downregulated in CBS138 VORI vs DMSO: %d", nrow(DEGdf_down2x_FDR5_cv))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_cv, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_cv, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_cv, "../data/deseq2/Candida_glabrata_CBS138/CBS138_VORI_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_cv, "../data/deseq2/Candida_glabrata_CBS138/CBS138_VORI_DMSO_down.csv")

```

```{r cbs_fv_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in CBS138 VORI vs FLUC: %d", nrow(DEGdf_up2x_FDR5_cvf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_cvf, n = 10)


sprintf("Downregulated in CBS138 VORI vs FLUC: %d", nrow(DEGdf_down2x_FDR5_cvf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_cvf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_cvf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_cvf, "../data/deseq2/Candida_glabrata_CBS138/CBS138_VORI_FLUC_up.csv")
write.csv(DEGdf_down2x_FDR5_cvf, "../data/deseq2/Candida_glabrata_CBS138/CBS138_VORI_FLUC_down.csv")

```

BG2:

```{r bg2_f_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in BG2 FLUC vs DMSO: %d", nrow(DEGdf_up2x_FDR5_bf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_bf, n = 10)


sprintf("Downregulated in BG2 FLUC vs DMSO: %d", nrow(DEGdf_down2x_FDR5_bf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_bf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_bf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_bf, "../data/deseq2/Candida_glabrata_BG2/BG2_FLUC_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_bf, "../data/deseq2/Candida_glabrata_BG2/BG2_FLUC_DMSO_down.csv")

```

```{r bg2_v_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in BG2 VORI vs DMSO: %d", nrow(DEGdf_up2x_FDR5_bv))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_bv, n = 10)


sprintf("Downregulated in BG2 VORI vs DMSO: %d", nrow(DEGdf_down2x_FDR5_bv))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_bv, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_bv, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_bv, "../data/deseq2/Candida_glabrata_BG2/BG2_VORI_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_bv, "../data/deseq2/Candida_glabrata_BG2/BG2_VORI_DMSO_down.csv")

```

```{r bg2_vf_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in BG2 VORI vs DMSO: %d", nrow(DEGdf_up2x_FDR5_bfv))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_bfv, n = 10)


sprintf("Downregulated in BG2 VORI vs DMSO: %d", nrow(DEGdf_down2x_FDR5_bfv))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_bfv, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_bfv, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_bfv, "../data/deseq2/Candida_glabrata_BG2/BG2_VORI_FLUC_up.csv")
write.csv(DEGdf_down2x_FDR5_bfv, "../data/deseq2/Candida_glabrata_BG2/BG2_VORI_FLUC_down.csv")

```

**Conclusions:**

- no difference betwee FLUC and VORI in both strains
- small but comparable number of up/downregulatedg genes in both strains
- the CBS138_FLUC outlier might impact the differential expression analysis, so I might need to repeat the results on the two more consistent replicates.

### Identification of corresponding genes in S. cerevisiae and between the two C.glabrata strains

Using curated orthologs between C.glabrata CBS138 and S.cerevisiae (downloaded from candidagenome.org), I identified standard and systematic _S. cerevisiae_ names for _C.glabrata_ genes up- and down-regulated in FLUC vs DMSO.
There is no need to do repeat it in VORI, because there are no significant DEG between these two conditions.

Orthologs are also identified between C.glabrata strains using Reciprocal Blast Best Hits.

```{r ortho_sc, eval=TRUE, echo=FALSE, results="hide", message=FALSE, warning=FALSE}
ortho_cgob = read.table("../data/cgob/C_glabrata_CBS138_S_cerevisiae_orthologs.txt", sep="\t", comment.char = "#", header=FALSE)
colnames(ortho_cgob) = c("systematic_cg", "standard_cg", "database_cg", "systematic_sc", "standard_sc", "database_sc")
head(ortho_cgob)

ortho_up = merge(DEGdf_up2x_FDR5_cf, ortho_cgob, by.x="gene", by.y="systematic_cg")
head(ortho_up)

ortho_down = merge(DEGdf_down2x_FDR5_cf, ortho_cgob, by.x="gene", by.y="systematic_cg")
head(ortho_down)

write.csv(ortho_up, "../data/deseq2/Candida_glabrata_CBS138/CBS138_FLUC_DMSO_up_ortho.csv")
write.csv(ortho_down, "../data/deseq2/Candida_glabrata_CBS138/CBS138_FLUC_DMSO_down_ortho.csv")
```

Here I look at overlap between genes upregulated/downregulated in FLUC treatment between CBS138 and BG2.
This includes:

- reporting the number of genes that are DE in one strain and have a homolog in the other - this shows how many genes unique for the strain are DE
- reporting the number of genes that are DE in both strain - this shows whether the response is similar between CBS138 and BG2.

```{r ortho_cb, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE}
rbbh = read.csv("../data/rbbh/cbs138_bg2_rbbh2.csv")
# rbbh = py$rbbh_cgla_clga_ids_gene
# write.csv(rbbh, "../data/rbbh/cbs138_bg2_rbbh.csv")
# head(rbbh)

ortho_cb_c_up = merge(DEGdf_up2x_FDR5_cf, rbbh, by.x="gene", by.y="gene_id_cbs138")
# head(ortho_cb_c_up)

ortho_cb_c_down = merge(DEGdf_down2x_FDR5_cf, rbbh, by.x="gene", by.y="gene_id_cbs138") # upregulated in CBS138 with a homolog in BG2
# head(ortho_cb_c_down)

ortho_cb_b_up = merge(DEGdf_up2x_FDR5_bf, rbbh, by.x="gene", by.y="gene_id_bg2")
# head(ortho_cb_b_up)

ortho_cb_b_down = merge(DEGdf_down2x_FDR5_bf, rbbh, by.x="gene", by.y="gene_id_bg2")
# head(ortho_cb_b_down)

sprintf("Number of FLUC upregulated genes in CBS138: %d that have a homolog in BG2: %d", nrow(DEGdf_up2x_FDR5_cf), nrow(ortho_cb_c_up))
sprintf("Number of FLUC downregulated genes in CBS138: %d that have a homolog in BG2: %d", nrow(DEGdf_down2x_FDR5_cf), nrow(ortho_cb_c_down))

sprintf("Number of FLUC upregulated genes in BG2: %d that have a homolog in CBS138: %d", nrow(DEGdf_up2x_FDR5_bf), nrow(ortho_cb_b_up))
sprintf("Number of FLUC downregulated genes in BG2: %d that have a homolog in CBS138: %d", nrow(DEGdf_down2x_FDR5_bf), nrow(ortho_cb_b_down))

ortho_cb_cb_up = merge(ortho_cb_c_up, ortho_cb_b_up, by.x = "gene", by.y = "gene_id_cbs138")
# head(ortho_cb_cb_up)

ortho_cb_cb_down = merge(ortho_cb_c_down, ortho_cb_b_down, by.x = "gene", by.y = "gene_id_cbs138")
# head(ortho_cb_cb_down)

sprintf("Number of FLUC upregulated genes overlapping between CBS138 and BG2: %d", nrow(ortho_cb_cb_up))
sprintf("Number of FLUC downregulated genes overlapping between CBS138 and BG2: %d", nrow(ortho_cb_cb_down))

write.csv(ortho_cb_c_up, "../data/ortho/CBS138_BG2_FLUC_DMSO_CBS138_up_ortho.csv")
write.csv(ortho_cb_c_down, "../data/ortho/CBS138_BG2_FLUC_DMSO_CBS138_down_ortho.csv")
write.csv(ortho_cb_b_up, "../data/ortho/CBS138_BG2_FLUC_DMSO_BG2_up_ortho.csv")
write.csv(ortho_cb_b_down, "../data/ortho/CBS138_BG2_FLUC_DMSO_BG2_down_ortho.csv")

write.csv(ortho_cb_cb_up, "../data/ortho/CBS138_BG2_FLUC_DMSO_common_up_ortho.csv")
write.csv(ortho_cb_cb_down, "../data/ortho/CBS138_BG2_FLUC_DMSO_common_down_ortho.csv")

```

**Conclusions:**

- most upregulated genes have homologs in both strains
- only a little more than 50% of genes differentially expressed in one strain are also differentially expressed in the other


### Gene Ontology analysis of up- and down-regulated genes in CBS138 and BG2

Based on this, I used GO term enrichment (candidagenome.org) of up- and downregulated genes in CBS138, BG2 and common to both strains.

The results are in ../data/go/ folder.

#### Upregulated genes

Both strains up (selected):

- secondary alcohol biosynthetic process
- sterol and lipid biosynthetic process
- sulfur compound metabolic process
- secondary alcohol metabolic process
- small molecule metabolic process
- alcohol biosynthetic process
- small molecule biosynthetic process
- organic hydroxy compound biosynthetic process
- homoserine metabolic process
- non-proteinogenic amino acid metabolic process
- sulfur amino acid biosynthetic process
- aspartate family amino acid metabolic process
- amino acid import across plasma membrane
- L-methionine biosynthetic process

Only CBS138 up:

- carbohydrate metabolic process
- oligosaccharide metabolic process
- monosaccharide metabolic process
- trehalose metabolic process

BG2 only up (selected):

- lysine biosynthetic process via aminoadipic acid
- proteinogenic amino acid metabolic process
- L-amino acid metabolic process
- aspartate family amino acid biosynthetic process


**Conclusions:**

- both strains have upregulation of ergosterol and lipid biosynthesis genes and small molecule metabolism genes (which also includes ERG genes)
- BG2 DEG are more enriched in amino acid metabolism genes
- CBS138 are more enriched in carbohydrate metabolism genes - I think these are generally upregulated in stress?

#### Downregulated genes

Both strains down (selected):

- ribosome biogenes
- nucleic acid catabolism

CBS138 only:

- arginine biosynthetic process
- arginine metabolic process
- glutamine family amino acid biosynthetic process
- ornithine metabolic process
- arginine biosynthetic process via ornithine
- glutamine family amino acid metabolic process
- ribosome biogenesis

BG2 only:

- translation
- ribosome biogenesis

**Conclusions:**

- in both strains mostly ribosome biogenesis genes are downregulated


### Comparison with transcription factor targets

Using data from two publications, both in CBS138:

- UPC2A https://doi.org/10.1371/journal.pgen.1009582
- PDR1 https://doi.org/10.1128/aac.03921-14

I looked at differential expression of two transcription factor targets involved in antifungal resistance.
This part of the analysis in CBS138 only.

This is not enrichment analysis, just search for overlapping genes.

For comparison, I'm also showing targets of _S. cerevisiae_ homologs of these two genes (targets downloaded from SGD).


```{r transcription_factors, echo=FALSE}
pdr1_targets_cg = read.csv("../data/supplementary_material_PDR1/table1.csv")
pdr1_targets_cg_up = merge(ortho_up, pdr1_targets_cg, by.x="gene", by.y="systematic_cg")
pdr1_targets_cg_down = merge(ortho_down, pdr1_targets_cg, by.x="gene", by.y="systematic_cg")
# head(pdr1_targets_cg_up)

sprintf("Number of PDR1 targets in C.glabrata: %d, number of upregulated genes: %d, number of downregulated genes: %d", nrow(pdr1_targets_cg), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
pdr1_targets_cg_up$standard_sc.x
print("Intersection DOWN:")
pdr1_targets_cg_down$standard_sc.x

pdr1_targets = read.csv("../data/transcription_factors/PDR1_targets.txt", sep="\t", comment.char="!")
# head(pdr1_targets)
pdr1_targets_up = merge(ortho_up, pdr1_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")
pdr1_targets_down = merge(ortho_down, pdr1_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")

sprintf("Number of PDR1 targets in S. cerevisiae: %d, , number of upregulated genes: %d, number of downregulated genes: %d", nrow(pdr1_targets), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
pdr1_targets_up$standard_sc
print("Intersection DOWN:")
pdr1_targets_down$standard_sc


upc2a_targets = read.csv("../data/supplementary_material_UPC2A/10_1371_journal_pgen_1009582/table_1b.csv")
upc2a_targets_up = merge(ortho_up, upc2a_targets, by.x="gene", by.y="Gene.ID")
upc2a_targets_down = merge(ortho_down, upc2a_targets, by.x="gene", by.y="Gene.ID")

sprintf("Number of UPC2A targets in C. glabrata: %d, , number of upregulated genes: %d, number of downregulated genes: %d", nrow(upc2a_targets), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
upc2a_targets_up$standard_sc
print("Intersection DOWN:")
upc2a_targets_down$standard_sc

upc2_targets = read.csv("../data/transcription_factors/UPC2_targets_manual.txt", sep="\t", comment.char="!")
# head(upc2_targets)
upc2_targets_up = merge(ortho_up, upc2_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")
upc2_targets_down = merge(ortho_down, upc2_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")

sprintf("Number of UPC2 targets in S. cerevisiae: %d, , number of upregulated genes: %d, number of downregulated genes: %d", nrow(upc2_targets), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
upc2_targets_up$standard_sc
print("Intersection DOWN:")
upc2_targets_down$standard_sc

```

**Conclusions:**

- I'm not sure how reliable this data is, for example most ERG genes do not show up among _C. glabrata_ UPC2A targets, but do show up among _S. cerevisiae_ Upc2; especially that the UPC2A datasets has more than 10x genes than UPC2.
- given that there is little overlap between CgPdr1 and ScPdr1 targets, I don't know if it's worth looking at S. cerevisiae orthologs
- using binding motifs might be a better idea.

## Differential gene expression comparison between the two strains

The genes between the two strains are matched using Reciprocal Blast Best Hits, which might not be entirely accurate given copy number variations etc. between the two species.
I have used orthofinder https://doi.org/10.1186/s13059-019-1832-y to find orthologs, but I haven't found a way to incorporate this into the analysis yet.


```{r rbbh, echo=FALSE, message=FALSE, warning=FALSE}
# rbbh = py$rbbh_cgla_clga_ids_gene
# head(rbbh)

counts_all_2_rbbh = merge(counts_all_2, rbbh, by.x="Geneid", by.y="gene_id_bg2")
# head(py$counts_all_2)
# head(counts_all_2_rbbh)
counts_all_2_rbbh["Geneid"] = counts_all_2_rbbh["gene_id_cbs138"]
counts_all_2_rbbh = counts_all_2_rbbh[, !(colnames(counts_all_2_rbbh) %in% c("gene_id_cbs138","gene_id_bg2"))]
#counts_all_2_rbbh.drop(["gene_id_cbs138", "gene_id_bg2"], axis=1, inplace=True)
# head(counts_all_2_rbbh)

counts_all_joined = merge(counts_all_1, counts_all_2_rbbh,by="Geneid")
head(counts_all_joined)

```

```{r c_b_d, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

samplesheet = read.table("../data/sample_sheets/DGE_samplesheet_canGla2.txt", header=TRUE)
# samplesheet = py$sample_sheet_df
counts_all = counts_all_joined
# head(counts_all)

conditions_to_compare = c("CBS138_RPMI_DMSO", "BG2_RPMI_DMSO")


sample_sheet_compared_conditions <-
  dplyr::filter(samplesheet,
                Condition %in% conditions_to_compare)
rownames(sample_sheet_compared_conditions) <- 1:nrow(sample_sheet_compared_conditions)

dds_compared_conditions <- 
  DESeqDataSetFromMatrix(countData = counts_all %>%
                           dplyr::select(sample_sheet_compared_conditions$Code) %>%
                           magrittr::set_rownames(counts_all$Gene),
                         colData = sample_sheet_compared_conditions,
                         design = ~ Strain) %>%
  DESeq()
dds_compared_conditions



results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)

deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_cbd = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cbd = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cbd = get_non_significant_genes(deseq_df)

print(DEGdf_up2x_FDR5_cbd, n = 20)

print(DEGdf_down2x_FDR5_cbd, n = 20)

print(DEGdf_non_significant_cbd, n = 20)

p7 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_cbd, DEGdf_down2x_FDR5_cbd) +
  ggtitle(paste0(c(unlist(strsplit(rN[2], "_")), "in RPMI DMSO"), collapse = " "))
#p7

print("Upregulated genes:")
nrow(DEGdf_up2x_FDR5_cbd)
print("Downregulated genes")
nrow(DEGdf_down2x_FDR5_cbd)
```

```{r c_b_v, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

samplesheet = read.table("../data/sample_sheets/DGE_samplesheet_canGla2.txt", header=TRUE)
# samplesheet = py$sample_sheet_df
counts_all = counts_all_joined
head(counts_all)

conditions_to_compare = c("CBS138_RPMI_VORI", "BG2_RPMI_VORI")


sample_sheet_compared_conditions <-
  dplyr::filter(samplesheet,
                Condition %in% conditions_to_compare)
rownames(sample_sheet_compared_conditions) <- 1:nrow(sample_sheet_compared_conditions)

dds_compared_conditions <- 
  DESeqDataSetFromMatrix(countData = counts_all %>%
                           dplyr::select(sample_sheet_compared_conditions$Code) %>%
                           magrittr::set_rownames(counts_all$Gene),
                         colData = sample_sheet_compared_conditions,
                         design = ~ Strain) %>%
  DESeq()
dds_compared_conditions



results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)

deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_cbv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cbv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cbv = get_non_significant_genes(deseq_df)

print(DEGdf_up2x_FDR5_cbv, n = 20)

print(DEGdf_down2x_FDR5_cbv, n = 20)

print(DEGdf_non_significant_cbv, n = 20)

p8 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_cbv, DEGdf_down2x_FDR5_cbv) +
  ggtitle(paste0(c(unlist(strsplit(rN[2], "_")), "in RPMI VORI"), collapse = " "))
#p8

print("Upregulated genes:")
nrow(DEGdf_up2x_FDR5_cbv)
print("Downregulated genes")
nrow(DEGdf_down2x_FDR5_cbv)
```

```{r c_b_f, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

samplesheet = read.table("../data/sample_sheets/DGE_samplesheet_canGla2.txt", header=TRUE)
# samplesheet = py$sample_sheet_df
counts_all = counts_all_joined
# head(counts_all)

conditions_to_compare = c("CBS138_RPMI_FLUC", "BG2_RPMI_FLUC")


sample_sheet_compared_conditions <-
  dplyr::filter(samplesheet,
                Condition %in% conditions_to_compare)
rownames(sample_sheet_compared_conditions) <- 1:nrow(sample_sheet_compared_conditions)

dds_compared_conditions <- 
  DESeqDataSetFromMatrix(countData = counts_all %>%
                           dplyr::select(sample_sheet_compared_conditions$Code) %>%
                           magrittr::set_rownames(counts_all$Gene),
                         colData = sample_sheet_compared_conditions,
                         design = ~ Strain) %>%
  DESeq()
dds_compared_conditions



results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)

deseq_df <- deseq_df_transform(dds_compared_conditions)

DEGdf_up2x_FDR5_cbf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cbf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cbf = get_non_significant_genes(deseq_df)

print(DEGdf_up2x_FDR5_cbf, n = 20)

print(DEGdf_down2x_FDR5_cbf, n = 20)

print(DEGdf_non_significant_cbf, n = 20)

p9 = volcano_plot(deseq_df, rN[2], DEGdf_up2x_FDR5_cbf, DEGdf_down2x_FDR5_cbf) +
  ggtitle(paste0(c(unlist(strsplit(rN[2], "_")), "in RPMI FLUC"), collapse = " "))
#p9

print("Upregulated genes:")
nrow(DEGdf_up2x_FDR5_cbf)
print("Downregulated genes")
nrow(DEGdf_down2x_FDR5_cbf)
```

```{r arrange_plots_2, fig.height=5.5, fig.width=8, echo=FALSE, message=FALSE, warning=FALSE}
pg = plot_grid(p7, p8, p9, NULL, ncol = 2, nrow = 2, rel_widths = c(5, 5))
pg
```

Report number of differentially expressed genes in CBS138 vs BG2:

```{r diff_print, echo=FALSE}

sprintf("DMSO upregulated: %d, downregulated: %d", nrow(DEGdf_up2x_FDR5_cbd), nrow(DEGdf_down2x_FDR5_cbd))

sprintf("FLUC upregulated: %d, downregulated: %d", nrow(DEGdf_up2x_FDR5_cbf), nrow(DEGdf_down2x_FDR5_cbf))

sprintf("VORI upregulated: %d, downregulated: %d", nrow(DEGdf_up2x_FDR5_cbv), nrow(DEGdf_down2x_FDR5_cbv))

write.csv(DEGdf_up2x_FDR5_cbd, "../data/deseq2/Candida_glabrata_both/CBS138_BG2_DMSO_up.csv")

write.csv(DEGdf_down2x_FDR5_cbd, "../data/deseq2/Candida_glabrata_both/CBS138_BG2_DMSO_down.csv")

write.csv(DEGdf_up2x_FDR5_cbf, "../data/deseq2/Candida_glabrata_both/CBS138_BG2_FLUC_up.csv")

write.csv(DEGdf_down2x_FDR5_cbf, "../data/deseq2/Candida_glabrata_both/CBS138_BG2_FLUC_down.csv")

write.csv(DEGdf_up2x_FDR5_cbv, "../data/deseq2/Candida_glabrata_both/CBS138_BG2_VORI_up.csv")

write.csv(DEGdf_down2x_FDR5_cbv, "../data/deseq2/Candida_glabrata_both/CBS138_BG2_VORI_down.csv")


```

### GO analysis of strain differences

DMSO upregulated in CBS138 (selected):

- amino acid biosynthesis
- cell adhesion
- spore wall assembly
- cell wall assembly

DMSO downregulated in CBS138 (selected):

- trehalose metabolism
- small molecule catabolism
- lipid catabolism

FLUC upregulated in CBS138:

- cell adhesion: AWP2, CAGL0A01584g, CAGL0I09856g, CAGL0K12078g, EPA23, EPA6

FLUC downregulated in CBS138 (selected):

- small molecule catabolism
- energy derivation by oxidation of organic compounds

VORI upregulated in CBS138:

- cell adhesion: AWP2, CAGL0A01584g, CAGL0I09856g, EPA23, EPA3, EPA6
- filamentous growth: CAGL0H02145g, CAGL0I01254g, CAGL0I09856g, CAGL0K04169g, CAGL0M03663g, CAGL0M09042g, EFG1, SHO1, TEC1, YPS2

VORI downregulated in CBS138:

- small molecule catabolism
- trehalose metabolism
- carbohydrate metabolism
- energy derivation by oxidation of organic compounds

## PCA

```{r pca_data}

samplesheet = here::here("data", 
                         "sample_sheets", 
                         "DGE_samplesheet_canGla2.txt") %>%
  read_table()
# samplesheet = py$experiment_data_both
rownames(samplesheet) = samplesheet$Code
print(samplesheet)

counts_only = subset(counts_all, select=samplesheet$Code)
#counts_only <- counts_all[,-1]
rownames(counts_only) <- counts_all[,1]

print(head(counts_only))

dds_all <- DESeqDataSetFromMatrix(countData = counts_only,
                                  colData = samplesheet,
                                  design = ~ Code)

rlog_all <- rlog(dds_all)
head(assay(rlog_all))
rlog_all_df = as.data.frame(assay(rlog_all))
  
```


```{r calculate_pca_rlog, eval=TRUE}
# calculate principal components of rlog, after extracting from the dataset
pca_rlog <- rlog_all %>%
  assay() %>%
  t() %>%
  prcomp()

# convert principal components to data frame
pcdf_rlog <- bind_cols(
  as_tibble(colData(rlog_all)),
  as_tibble(pca_rlog$x)
)

pcdf_rlog
```

```{r calculate_propvar, eval=TRUE}
propvar_rlog_df <- tibble(
  PC = seq.int(1L, ncol(pca_rlog$x) ),
  tot_var = pca_rlog$sdev^2,
  prop_var = tot_var/sum(tot_var)
)

propvar_rlog_df
```

```{r plot_percentvar, fig.width = 3, fig.height = 2, eval=TRUE}
plot_percentvar_rlog <- 
  ggplot(data = propvar_rlog_df, 
         aes(x = PC, y = prop_var)) +
  geom_col(fill = "blue") +
  scale_x_continuous("principal component",
                     limits = c(0.4,10.6), 
                     breaks = 1L:10L,
                     expand = c(0,0)) + 
  scale_y_continuous("prop. of variance", expand = c(0,0))
plot_percentvar_rlog
```

```{r plot_PC12_default, fig.height=4, fig.width=7, eval=TRUE}
scales_PCA <- 
  list(
    scale_colour_manual("Strain, Treatment",
      values = c("CBS138_RPMI_DMSO" = "#5B2C6F",
                 "CBS138_RPMI_FLUC" = "#A569BD",
                 "CBS138_RPMI_VORI" = "#D2B4DE",
                 "BG2_RPMI_DMSO" = "#196F3D",
                 "BG2_RPMI_FLUC" = "#27AE60",
                 "BG2_RPMI_VORI" = "#A9DFBF"),
      labels = c("CBS138_RPMI_DMSO" = "CBS138, DMSO",
                 "CBS138_RPMI_FLUC" = "CBS138, FLUC",
                 "CBS138_RPMI_VORI" = "CBS138, VORI",
                 "BG2_RPMI_DMSO"    = "BG2,    DMSO",
                 "BG2_RPMI_FLUC"    = "BG2,    FLUC",
                 "BG2_RPMI_VORI"    = "BG2,    VORI")),
    scale_shape_manual("Strain, Treatment",
      values = c("CBS138_RPMI_DMSO" = 15,
                 "CBS138_RPMI_FLUC" = 15,
                 "CBS138_RPMI_VORI" = 15,
                 "BG2_RPMI_DMSO" = 17,
                 "BG2_RPMI_FLUC" = 17,
                 "BG2_RPMI_VORI" = 17),
      labels = c("CBS138_RPMI_DMSO" = "CBS138, DMSO",
                 "CBS138_RPMI_FLUC" = "CBS138, FLUC",
                 "CBS138_RPMI_VORI" = "CBS138, VORI",
                 "BG2_RPMI_DMSO"    = "BG2,    DMSO",
                 "BG2_RPMI_FLUC"    = "BG2,    FLUC",
                 "BG2_RPMI_VORI"    = "BG2,    VORI"))
  )


pc12 = ggplot(data = pcdf_rlog,
       aes(colour = Condition, shape = Condition)
       ) +
  geom_point(aes(x = PC1, y = PC2)) +
  theme(legend.box = "horizontal") +
  scales_PCA
legend = get_legend(pc12)
pc12

pc32 = ggplot(data = pcdf_rlog,
       aes(colour = Condition, shape = Condition)
       ) +
  geom_point(aes(x = PC3, y = PC2)) +
  scales_PCA
pc32

ggplot(data = pcdf_rlog,
       aes(colour = Condition, shape = Rep)
       ) +
  geom_point(aes(x = PC3, y = PC2)) +
  scale_shape_identity()

# TODO: change styling to include shapes: circles squares triangles, maybe add Vs and Fs
```

Conclusions:

In PC1 vs PC2 there are 4 clear clusters: corresponding to treated and untreated strains BG2 and CBS138.
In PC3 they look more similar.

The outlier: CBS138_RPMI_VORI_C (light purple) clusters with other bio reps in PC1-PC2, but not in PC3.
After discussion, we decided that something was not quite right in that replicate.


## Venn Diagram of DEG lists

```{r venn}
library(VennDiagram)
 
# Generate 3 sets of 200 words
set1 <- paste(rep("word_" , 200) , sample(c(1:1000) , 200 , replace=F) , sep="")
set2 <- paste(rep("word_" , 200) , sample(c(1:1000) , 200 , replace=F) , sep="")
set3 <- paste(rep("word_" , 200) , sample(c(1:1000) , 200 , replace=F) , sep="")
 
set.seed(20190708)
genes <- paste("gene",1:1000,sep="")
x <- list(
  A = sample(genes,300), 
  B = sample(genes,525), 
  C = sample(genes,440),
  D = sample(genes,350)
  )

# Chart
venn.diagram(
  x = list(set1, set2, set3),
  category.names = c("Set 1" , "Set 2 " , "Set 3"),
  filename = NULL,
  output=TRUE
)

display_venn <- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}
display_venn(
  x,
  category.names = c("Set 1" , "Set 2 " , "Set 3", "Set 4"),
  fill = c("#999999", "#E69F00", "#56B4E9", "#009E73")
  )
display_venn(x)

library("ggVennDiagram")
ggVennDiagram(x, label_alpha = 0)

cf_up = as.vector(ortho_cb_c_up$gene)
cbd_up = as.vector(DEGdf_up2x_FDR5_cbd$gene)
cbf_up = as.vector(DEGdf_up2x_FDR5_cbf$gene)
bf_up = as.vector(ortho_cb_b_up$gene_id_cbs138)

y <- list(
  CBS138_FLUC = cf_up, 
  strain_DMSO = cbd_up, 
  strain_FLUC = cbf_up,
  BG2_FLUC = bf_up
  )

display_venn(
  y,
  fill = c("#999999", "#E69F00", "#56B4E9", "#009E73")
  )


print(head(DEGdf_up2x_FDR5_cbd))
```


```{python interpretation_figures_rlog, fig.height = 2, fig.width=2, eval=FALSE}
import pandas as pd

plt.rcParams['svg.fonttype'] = 'none'

rlog_all_p = r.rlog_all_df
rlog_all_p = rlog_all_p.reset_index().rename(columns={'index': 'Geneid'})


filamentous_genes = ["CAGL0H02145g",	"CAGL0I01254g",	"CAGL0I09856g", "CAGL0K04169g", "CAGL0M03663g", "CAGL0M09042g"]
cell_adhesion_genes = ["CAGL0K00110g", "CAGL0A01584g", "CAGL0I09856g", "CAGL0K12078g", "CAGL0I00220g", "CAGL0E06688g", "CAGL0C00110g"]

s = rlog_all_p[rlog_all_p["Geneid"].isin(cell_adhesion_genes)]

t = s.set_index("Geneid")
#plt.figure(figsize=(10, 10))
#plt.subplots_adjust(left=0.5, right=1.5, bottom=4.0)
g = sns.clustermap(t, cmap='vlag', method="single", annot_kws={"size": 7})
plt.setp(g.ax_heatmap.get_yticklabels(), rotation=0)  # For y axis
plt.show()
#plt.savefig("/Users/s1964600/Work/random_figures/new_svg/cpac_scer_deg.svg")
plt.close()

```


TODOs:

- [ ] look at the orthogroups for differences between the species
- [ ] change data formatting to have one file with orthology information instead of multiple files for every condition
- [ ] extend the analysis of regulation, including the strain differences
- [ ] less text, more figures
