---
title: "Quality Control and Differential Gene Expression Analysis, Candida glabrata"
author: "Weronika Danecka and Edward Wallace"
date: "2024-05-23"
output: 
  html_document:
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
options(knitr.duplicate.label = "allow")
knitr::opts_chunk$set(echo = TRUE)

library(tidyr)
library(tidyverse)
library(cowplot)
library(dplyr)
library(DESeq2)
library(GGally)
library(ggrepel)
library(here)
library(biobroom)

theme_set(theme_cowplot(font_size = 12) +
            theme(strip.background = element_blank(),
                  axis.text=element_text(size=10),
                  axis.title=element_text(size=10),
                  plot.title = element_text(size=10),
                  legend.title=element_text(size=10),
                  legend.text=element_text(size=10)))

COUNT_HEADER_COLUMNS <- c("Geneid", "Chr", "Start",
                          "End", "Strand", "Length")

#knitr::knit_child("featureCounts_preprocess.Rmd")
#knitr::knit_child("ortho_preprocess.Rmd")
```

# Introduction

Candida glabrata strains:

1. CBS138
2. BG2

Treated with voriconazole (VCZ / VORI), fluconazole (FCZ / FLUC) or DMSO (solvent control).

Sequencing FASTQ files were aligned to appropriate genomes using HISAT2 and quantified using featureCounts.

MultiQC results, including quality control of raw reads (FastQC) and alignment are here:

- CBS138: `r paste('..','data', 'multiQC','Candida_glabrata_CBS138','multiqc_report.html', sep="/")`)
- BG2: `r paste('..','data', 'multiQC','Candida_glabrata_BG2','multiqc_report.html', sep="/")`)

featureCount results, aggregated by strain, are below:

```{r show_data_1}
counts_all_CBS138 = read.table("../data/featureCounts/Candida_glabrata_CBS138/CBS138_all_conditions_featureCounts.tab", header=TRUE)
counts_all_BG2 = read.table("../data/featureCounts/Candida_glabrata_BG2/BG2_all_conditions_featureCounts.tab", header=TRUE)

counts_rpm_CBS138 = read.table("../data/featureCounts/Candida_glabrata_CBS138/CBS138_all_conditions_rpm.tab", header=TRUE)
counts_rpm_BG2 = read.table("../data/featureCounts/Candida_glabrata_BG2/BG2_all_conditions_rpm.tab", header=TRUE)

head(counts_all_CBS138) # CBS138, raw
head(counts_all_BG2) # BG2, raw
```

Results normalised to RPM are below:

```{r show_data}

head(counts_rpm_CBS138) # CBS138, normalised to RPM
head(counts_rpm_BG2) # BG2, normalised to RPM

```

## Match genes across strains using Reciprocal Blast Best Hits

The genes between the two strains are matched using Reciprocal Blast Best Hits, which might not be entirely accurate given copy number variations etc. between the two species.
I have used orthofinder https://doi.org/10.1186/s13059-019-1832-y to find orthologs, but I haven't found a way to incorporate this into the analysis yet.


```{r rbbh, echo=TRUE, message=FALSE, warning=FALSE}
# Load reciprocal BLAST best hits
rbbh = read.csv("../data/rbbh/cbs138_bg2_rbbh2.csv")

# merge on CBS138 gene names,
counts_all_CBS138_rbbh = merge(counts_all_CBS138, rbbh, 
                               by.x="Geneid", by.y="gene_id_cbs138")

# combine counts from common genes.
counts_all_joined = merge(counts_all_CBS138_rbbh, 
                          dplyr::rename(counts_all_BG2, gene_id_bg2 = "Geneid"), 
                          by = "gene_id_bg2")
head(counts_all_joined)

```


# Analysis

## Quality control - sample correlation

For quality control, we check correlation between the samples (only within strain).

There is high correlation within untreated (DMSO) samples and treated (FCZ and VCZ), but no obvious differences between VCZ and FCZ.

```{r pairplot_counts, echo = FALSE, fig.height = 8, fig.width = 8}
# set up short sample codes
shortcodes <- 
  paste(rep(c("DMSO", "FLUC", "VORI"), each = 3), c("A","B","C"),
        sep = "_")

# define maximum count so 
maxcount <- max( counts_all_CBS138 %>%
                   dplyr::select(-Geneid) %>% max(),
                 counts_all_BG2 %>%
                   dplyr::select(-Geneid) %>% max())
maxlog10count <- log10(maxcount + 100)

# function to format ggpairs nicely
pair_points <- function(data, mapping, ...){
    ggplot(data=data) + 
      geom_abline(mapping = NULL,
                  intercept = 0, slope = 1, linetype = "dashed") + 
      geom_point(mapping=mapping, colour = "grey50", size = 0.5) + 
      scale_y_continuous(breaks = 0:6, limits = c(0,maxlog10count)) +
      scale_x_continuous(breaks = 0:6, limits = c(0,maxlog10count))
  }

ggpairs(data = counts_all_CBS138 %>%
          dplyr::rename_with(.fn = stringr::str_remove, pattern = "CBS138_RPMI_") %>%
          dplyr::select(all_of(shortcodes)) %>%
          magrittr::add(1) %>%
          log10(),
        lower = list(continuous = pair_points),
        title = "Strain CBS138, log10(counts + 1)")

ggpairs(data = counts_all_BG2 %>%
          dplyr::rename_with(.fn = stringr::str_remove, pattern = "BG2_RPMI_") %>%
          dplyr::select(all_of(shortcodes)) %>%
          magrittr::add(1) %>%
          log10(),
        lower = list(continuous = pair_points),
        title = "Strain BG2, log10(counts + 1)")

```

This shows that the replicates are generally very well correlated, R > 0.96. However, there is more variability in replicate C, especially CBS138_RPMI_VORI_C.


## Quality control - clustering

We intially cluster the results on raw rpm, separately by strain. Again untreated samples cluster together and treated samples cluster together.
Treated samples cluster in a way that doesn't indicate any obvious sample mix-up.

CBS138_RPMI_VORI_C is an outlier.


```{r heatmap_rpm, fig.height = 8, fig.width=8, echo=FALSE, message=FALSE}
library(pheatmap)

# create heatmap using pheatmap
ph1 = pheatmap(counts_rpm_CBS138[,-1], main="CBS138", show_rownames = FALSE)
ph2 = pheatmap(counts_rpm_BG2[,-1], main="BG2", show_rownames = FALSE)
```


### Calculate rlog

Regularized logarithm, on both strains together joined by best-hit genes.

```{r calculate_rlog_all}
sample_sheet_all = here::here("data", 
                         "sample_sheets", 
                         "DGE_samplesheet_canGla2.txt") %>%
  read_table()
# samplesheet = py$experiment_data_both
rownames(sample_sheet_all) = sample_sheet_all$Code
print(sample_sheet_all)

dds_all <- DESeqDataSetFromMatrix(countData =
                                    counts_all_joined %>%
                             dplyr::select(sample_sheet_all$Code) %>%
                             magrittr::set_rownames(counts_all_joined$Gene),
                                  colData = sample_sheet_all,
                                  design = ~ Code)

rlog_all <- rlog(dds_all)
head(assay(rlog_all))
rlog_all_df = as.data.frame(assay(rlog_all))
```

### Clustering on rlog across strains

```{r heatmap_rlog, fig.height = 8, fig.width=8, echo=FALSE, message=FALSE}

ph_rlog = pheatmap(rlog_all_df, 
                   main="Both strains, rlog", 
                   show_rownames = FALSE)
```

### Clustering on normalized rlog across strains

Normalized rlog gets at relative (not absolute) differences in expression between strains.
Here we additionally filter for a minimal count of 100, to remove low-count noise.


```{r heatmap_rlog_norm, fig.height = 8, fig.width=8, echo=FALSE, message=FALSE}

mincounts <- 
  # list of genes with
  counts_all_joined %>%
  dplyr::select(all_of(sample_sheet_all$Code)) %>%
  as.matrix() %>%
  rowMins()
genelist_mincount100 <- counts_all_joined$Geneid[mincounts >= 100]

ph_rlog = pheatmap(rlog_all_df[genelist_mincount100,] -
                     rowMeans(rlog_all_df[genelist_mincount100,]),
                   main="Both strains, normalized rlog, min. count 100", 
                   show_rownames = FALSE)
```

These clustering analyses support:

- clustering by - / + drug dominates clustering by strain
- the difference between the two drugs is minimal
- within drug treated samples, strains cluster.
- CBS138_RPMI_VORI_C is an outlier

We are not trying to get any more information out of this analysis, e.g. not trying to identify clusters of co-regulated genes, so there is no need to do more arranging the plots.


## Quality Control - Principal Component Analysis

Calculate principal components from rlog.

```{r calculate_pca_rlog, eval=TRUE}
# calculate principal components of rlog, after extracting from the dataset
pca_rlog <- rlog_all %>%
  assay() %>%
  t() %>%
  prcomp()

# convert principal components to data frame
pcdf_rlog <- bind_cols(
  as_tibble(colData(rlog_all)),
  as_tibble(pca_rlog$x)
)

pcdf_rlog
```

```{r calculate_propvar, eval=TRUE}
propvar_rlog_df <- tibble(
  PC = seq.int(1L, ncol(pca_rlog$x) ),
  tot_var = pca_rlog$sdev^2,
  prop_var = tot_var/sum(tot_var)
)

propvar_rlog_df
```

```{r plot_percentvar, fig.width = 3, fig.height = 2, eval=TRUE}
plot_percentvar_rlog <- 
  ggplot(data = propvar_rlog_df, 
         aes(x = PC, y = prop_var)) +
  geom_col(fill = "skyblue3") +
  scale_x_continuous("principal component",
                     limits = c(0.4,10.6), 
                     breaks = 1L:10L,
                     expand = c(0,0)) + 
  scale_y_continuous("prop. of variance", expand = c(0,0))
plot_percentvar_rlog
```

```{r plot_PC12_default, fig.height=4, fig.width=7, eval=TRUE}
scales_PCA <- 
  list(
    scale_colour_manual("Strain, Treatment",
      values = c("CBS138_RPMI_DMSO" = "grey20",
                 "CBS138_RPMI_FLUC" = "purple3",
                 "CBS138_RPMI_VORI" = "green4",
                 "BG2_RPMI_DMSO" = "grey20",
                 "BG2_RPMI_FLUC" = "purple3",
                 "BG2_RPMI_VORI" = "green4"),
      labels = c("CBS138_RPMI_DMSO" = "CBS138, DMSO",
                 "CBS138_RPMI_FLUC" = "CBS138, FCZ",
                 "CBS138_RPMI_VORI" = "CBS138, VCZ",
                 "BG2_RPMI_DMSO"    = "BG2, DMSO",
                 "BG2_RPMI_FLUC"    = "BG2, FCZ",
                 "BG2_RPMI_VORI"    = "BG2, VCZ")),
    scale_shape_manual("Strain, Treatment",
      values = c("CBS138_RPMI_DMSO" = 16,
                 "CBS138_RPMI_FLUC" = 16,
                 "CBS138_RPMI_VORI" = 16,
                 "BG2_RPMI_DMSO" = 17,
                 "BG2_RPMI_FLUC" = 17,
                 "BG2_RPMI_VORI" = 17),
      labels = c("CBS138_RPMI_DMSO" = "CBS138, DMSO",
                 "CBS138_RPMI_FLUC" = "CBS138, FCZ",
                 "CBS138_RPMI_VORI" = "CBS138, VCZ",
                 "BG2_RPMI_DMSO"    = "BG2, DMSO",
                 "BG2_RPMI_FLUC"    = "BG2, FCZ",
                 "BG2_RPMI_VORI"    = "BG2, VCZ"))
  )


pc12 = ggplot(data = pcdf_rlog,
       aes(colour = Condition, shape = Condition)
       ) +
  geom_hline(yintercept = 0, colour = "grey90") +
  geom_vline(xintercept = 0, colour = "grey90") +
  geom_point(aes(x = PC1, y = PC2), size = 2) +
  theme(legend.box = "horizontal") +
  scales_PCA
legend = get_legend(pc12)
pc12

pc32 = ggplot(data = pcdf_rlog,
       aes(colour = Condition, shape = Condition)
       ) +
  geom_hline(yintercept = 0, colour = "grey90") +
  geom_vline(xintercept = 0, colour = "grey90") +
  geom_point(aes(x = PC3, y = PC2), size = 2) +
  scales_PCA
pc32

ggplot(data = pcdf_rlog,
       aes(colour = Condition, shape = Rep)
       ) +
  geom_point(aes(x = PC3, y = PC2)) +
  scale_shape_identity()

# TODO: change styling to include shapes: circles squares triangles, maybe add Vs and Fs
```

Conclusions:

In PC1 vs PC2 there are 4 clear clusters: corresponding to treated and untreated strains BG2 and CBS138.
In PC3 they look more similar.

The outlier: CBS138_RPMI_VORI_C (light purple) clusters with other bio reps in PC1-PC2, but not in PC3.
After discussion, we decided that something was not quite right in that replicate.



## Differential Gene Expression analysis using DESeq2 in each strain separately

Differential Gene Expression analysis was performed using DESeq2.
Genes with adjusted p-value < 0.05 and at least 2x change are considered differentially expressed here.
The upregulated and downregulated genes are saved in ../data/deseq2


```{r functions, echo = FALSE}
# wrapper functions for DESeq2 package functions

run_deseq_vsmedium <- 
  function(counts_all,
           conditions_to_compare,
           strains_to_include = c("BG2", "CBS138"),
           sample_sheet = sample_sheet_all) {
    sample_sheet_compared_conditions <-
      dplyr::filter(sample_sheet,
                    Condition %in% conditions_to_compare,
                    Strain %in% strains_to_include)
    rownames(sample_sheet_compared_conditions) <-
      1:nrow(sample_sheet_compared_conditions)
    
    dds_compared_conditions <- 
      DESeqDataSetFromMatrix(
        countData = counts_all %>%
          dplyr::select(sample_sheet_compared_conditions$Code) %>%
          magrittr::set_rownames(counts_all$Gene),
        colData = sample_sheet_compared_conditions,
        design = ~ Medium) %>%
      DESeq()
    dds_compared_conditions
  }

run_deseq_vsstrain <- 
  function(counts_all,
           conditions_to_compare,
           sample_sheet = sample_sheet_all) {
    sample_sheet_compared_conditions <-
      dplyr::filter(sample_sheet,
                    Condition %in% conditions_to_compare)
    rownames(sample_sheet_compared_conditions) <-
      1:nrow(sample_sheet_compared_conditions)
    
    dds_compared_conditions <- 
      DESeqDataSetFromMatrix(
        countData = counts_all %>%
          dplyr::select(sample_sheet_compared_conditions$Code) %>%
          magrittr::set_rownames(counts_all$Gene),
        colData = sample_sheet_compared_conditions,
        design = ~ Strain) %>%
      DESeq()
    dds_compared_conditions
  }

deseq_df_transform <- function(dds_compared_conditions, 
                               termstokeep) {
  deseq_df <- 
  biobroom::tidy.DESeqDataSet(dds_compared_conditions) %>%
  dplyr::filter(term %in% termstokeep) %>%
  dplyr::select(gene, 
                baseMean, 
                log2FC = estimate, 
                stderror,
                padj = p.adjusted)
  deseq_df
}

get_upregulated_genes <- function(deseq_df) {
  DEGdf_up2x_FDR5 <- 
    deseq_df %>%
    dplyr::filter(log2FC > 1, padj < 0.05) %>%
    dplyr::arrange(desc(log2FC))
  DEGdf_up2x_FDR5
}

get_downregulated_genes <-function(deseq_df) {
  DEGdf_down2x_FDR5 <- 
    deseq_df %>%
    dplyr::filter(log2FC < -1, padj < 0.05) %>%
    dplyr::arrange(log2FC)
    
  DEGdf_down2x_FDR5
}
  
get_non_significant_genes <- function(deseq_df) {
  # logic here in "filter" call is bad. 
  DEGdf_non_significant <- 
    deseq_df %>%
    dplyr::filter(log2FC > -1, , log2FC < 1, padj > 0.05) %>%
    dplyr::arrange(log2FC)
  DEGdf_non_significant
}

volcano_plot <- function(deseq_df, up, down, 
                         xlabel = "log2 fold-change",
                         xlimits = c(-4,4),
                         ylimits = c(0,50)) {
  ggplot(data = deseq_df,
         aes(x = log2FC, y = -log10(padj)) ) +
    geom_point(size = 0.5, colour = "grey50") +
    geom_point(data = up,
               size = 1, colour = "darkblue") +
    geom_point(data = down,
               size = 1, colour = "darkred") +
    scale_y_continuous("-log10(p)",
                       limits = ylimits, expand = c(0,0),
                       oob=scales::squish) +
    scale_x_continuous(xlabel,
                       breaks = -5:5,
                       limits = xlimits, expand = c(0.01,0.01),
                       oob = scales::squish) +
    coord_cartesian(clip = "off") +
    theme(plot.title = element_text(hjust = 0.5))
}


```

### DGE, CBS138, FCZ vs DMSO

```{r cbs_f_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_CBS138,
                     conditions_to_compare = c("CBS138_RPMI_DMSO", "CBS138_RPMI_FLUC"),
                     strains_to_include = "CBS138")

dds_compared_conditions

results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_cf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/CBS138/CBS138_FLUC_DMSO_allDESeq.csv")


p_cf = volcano_plot(deseq_df, DEGdf_up2x_FDR5_cf, DEGdf_down2x_FDR5_cf,
                  xlabel = "log2 fold-change\n← down in FCZ       up in FCZ →") +
  ggtitle("FCZ vs DMSO, Strain CBS138")
#p_cf
```


### DGE, CBS138, VCZ vs DMSO

```{r cbs_v_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_CBS138,
                     conditions_to_compare = c("CBS138_RPMI_DMSO", "CBS138_RPMI_VORI"),
                     strains_to_include = "CBS138")
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_cv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cv = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/CBS138/CBS138_VORI_DMSO_allDESeq.csv")


p_cv = volcano_plot(deseq_df, DEGdf_up2x_FDR5_cv, DEGdf_down2x_FDR5_cv,
                  xlabel = "log2 fold-change\n← down in VCZ       up in VCZ →") +
  ggtitle("VCZ vs DMSO, Strain CBS138")
#p_cv

```

### Check: DGE, CBS138, FCZ vs DMSO excluding the outlying sample CBS138_RPMI_VORI_C

Conclude: the single outlying sample `CBS138_RPMI_VORI_C` has minimal effect on the substantial results. Although it changes the genelists over the cutoff, it doesn't change much most of the fold changes and p-values.

```{r cbs_v_deseq_excludebadc, fig.height=4, fig.width=5, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_CBS138,
                     conditions_to_compare = c("CBS138_RPMI_DMSO", "CBS138_RPMI_VORI"),
                     strains_to_include = "CBS138",
                     sample_sheet = dplyr::filter(sample_sheet_all,
                                                  Code != "CBS138_RPMI_VORI_C"))
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df_xc <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_cvxc = get_upregulated_genes(deseq_df_xc)
DEGdf_down2x_FDR5_cvxc = get_downregulated_genes(deseq_df_xc)
DEGdf_non_significant_cvxc = get_non_significant_genes(deseq_df_xc)

# write.csv(deseq_df_xc, "../data/deseq2/CBS138/CBS138_VORI_DMSO_allDESeq.csv")


p_cvx = volcano_plot(deseq_df_xc, DEGdf_up2x_FDR5_cvxc, DEGdf_down2x_FDR5_cvxc,
                    xlabel = "log2 fold-change\n← down in VCZ       up in VCZ →") +
  ggtitle("VCZ vs DMSO, Strain CBS138, exclude outlying sample")
#p_cvx

deseq_df_3repsvsxc <- 
  dplyr::inner_join(deseq_df, 
              deseq_df_xc, 
              by = "gene",
              suffix = c(".all", ".xc"))

ggplot(data = deseq_df_3repsvsxc, aes(x = log2FC.all, y = log2FC.xc)) + 
  geom_hline(yintercept = c(-1, 1), colour = "purple", linetype = "dashed") + 
  geom_vline(xintercept = c(-1, 1), colour = "purple", linetype = "dashed") +
  geom_abline(intercept = 0, slope = 1, colour = "purple") + 
    geom_point(size = 0.2, colour = "grey50") +
    geom_point(data = dplyr::filter(deseq_df_3repsvsxc,
                             gene %in% DEGdf_up2x_FDR5_cvxc$gene),
               size = 1, colour = "darkblue") +
    geom_point(data = dplyr::filter(deseq_df_3repsvsxc,
                             gene %in% DEGdf_down2x_FDR5_cvxc$gene),
               size = 1, colour = "darkred")
```


### DGE, CBS138, VCZ vs FCZ

```{r cbs_vf_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_CBS138,
                     conditions_to_compare = c("CBS138_RPMI_FLUC", "CBS138_RPMI_VORI"),
                     strains_to_include = "CBS138")

dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_cvf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cvf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cvf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/CBS138/CBS138_VORI_FLUC_allDESeq.csv")

p_cvf = volcano_plot(deseq_df, DEGdf_up2x_FDR5_cvf, DEGdf_down2x_FDR5_cvf,
                    xlabel = "log2 fold-change\n← up in FCZ       up in VCZ →") +
  ggtitle("VCZ vs FCZ, Strain CBS138")
#p_cvf

```

### Double-check: DGE, CBS138, VCZ vs FCZ excluding the outlying sample CBS138_RPMI_VORI_C 


```{r cbs_vf_deseq_excludebadc, fig.height=4, fig.width=5, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_CBS138,
                     conditions_to_compare = c("CBS138_RPMI_FLUC", "CBS138_RPMI_VORI"),
                     strains_to_include = "CBS138",
                     sample_sheet = dplyr::filter(sample_sheet_all,
                                                  Code != "CBS138_RPMI_VORI_C"))

dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_cvfxc = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cvfxc = get_downregulated_genes(deseq_df)

volcano_plot(deseq_df, DEGdf_up2x_FDR5_cvfxc, DEGdf_down2x_FDR5_cvfxc,
                    xlabel = "log2 fold-change\n← up in FCZ       up in VCZ →") +
  ggtitle("VCZ vs FCZ, Strain CBS138")

```

Again this does not substantially change outcome: 7 DE genes when the sample is excluded, instead of zero.


### DGE, BG2, FCZ vs DMSO

```{r bg2_f_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_BG2,
                     conditions_to_compare = c("BG2_RPMI_DMSO", "BG2_RPMI_FLUC"),
                     strains_to_include = "BG2")

dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_bf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_bf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_bf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/BG2/BG2_FLUC_DMSO_allDESeq.csv")

p_bf = volcano_plot(deseq_df, DEGdf_up2x_FDR5_bf, DEGdf_down2x_FDR5_bf,
                    xlabel = "log2 fold-change\n← down in FCZ       up in FCZ →") +
  ggtitle("FCZ vs DMSO, Strain BG2")
#p_bf
```


### DGE, BG2, VCZ vs DMSO

```{r bg2_v_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_BG2,
                     conditions_to_compare = c("BG2_RPMI_DMSO", "BG2_RPMI_VORI"),
                     strains_to_include = "BG2")
dds_compared_conditions


results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_bv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_bv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_bv = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/BG2/BG2_VORI_DMSO_allDESeq.csv")

p_bv = volcano_plot(deseq_df, DEGdf_up2x_FDR5_bv, DEGdf_down2x_FDR5_bv,
                    xlabel = "log2 fold-change\n← down in VCZ       up in VCZ →") +
  ggtitle("VCZ vs DMSO, Strain BG2")
#p_bv
```


### DGE, BG2, VCZ vs FCZ

```{r bg2_fv_deseq, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, results='hide', message=FALSE, warning=FALSE}

dds_compared_conditions = 
  run_deseq_vsmedium(counts_all_BG2,
                     conditions_to_compare = c("BG2_RPMI_FLUC", "BG2_RPMI_VORI"),
                     strains_to_include = "BG2")

dds_compared_conditions

results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)
dds_compared_conditions
biobroom::tidy.DESeqDataSet(dds_compared_conditions)
deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])

DEGdf_up2x_FDR5_bvf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_bvf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_bvf = get_non_significant_genes(deseq_df)

write.csv(deseq_df, "../data/deseq2/BG2/BG2_VORI_FLUC_allDESeq.csv")

p_bvf = volcano_plot(deseq_df, DEGdf_up2x_FDR5_bvf, DEGdf_down2x_FDR5_bvf,
                    xlabel = "log2 fold-change\n← up in FCZ       up in VCZ →") +
  ggtitle("VCZ vs FCZ, Strain BG2")
# p_bvf
```

### Volcano plots of pairwise comparisons

```{r arrange_pairwise_volcano_plots, fig.height=7.5, fig.width=7, echo=FALSE, warning=FALSE}
pg = plot_grid(p_cf, p_bf, p_cv, p_bv, p_cvf, p_bvf, 
               ncol = 2, nrow = 3)
pg

ggsave("../figures/pairwise_volcano_plots.png", 
       plot = pg,
       width = 7.5, height = 7)
```

How many genes were differentially expressed in each condition? [TODO: make it a figure]

CBS138:

```{r cbs_f_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated CBS138 FCZ vs DMSO: %d", nrow(DEGdf_up2x_FDR5_cf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_cf, n = 10)


sprintf("Downregulated CBS138 FCZ vs DMSO: %d", nrow(DEGdf_down2x_FDR5_cf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_cf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_cf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_cf, "../data/deseq2/CBS138/CBS138_FLUC_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_cf, "../data/deseq2/CBS138/CBS138_FLUC_DMSO_down.csv")

```

```{r cbs_v_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in CBS138 VCZ vs DMSO: %d", nrow(DEGdf_up2x_FDR5_cv))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_cv, n = 10)


sprintf("Downregulated in CBS138 VCZ vs DMSO: %d", nrow(DEGdf_down2x_FDR5_cv))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_cv, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_cv, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_cv, "../data/deseq2/CBS138/CBS138_VORI_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_cv, "../data/deseq2/CBS138/CBS138_VORI_DMSO_down.csv")

```

```{r cbs_fv_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in CBS138 VCZ vs FCZ: %d", nrow(DEGdf_up2x_FDR5_cvf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_cvf, n = 10)


sprintf("Downregulated in CBS138 VCZ vs FCZ: %d", nrow(DEGdf_down2x_FDR5_cvf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_cvf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_cvf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_cvf, "../data/deseq2/CBS138/CBS138_VORI_FLUC_up.csv")
write.csv(DEGdf_down2x_FDR5_cvf, "../data/deseq2/CBS138/CBS138_VORI_FLUC_down.csv")

```

BG2:

```{r bg2_f_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in BG2 FCZ vs DMSO: %d", nrow(DEGdf_up2x_FDR5_bf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_bf, n = 10)


sprintf("Downregulated in BG2 FCZ vs DMSO: %d", nrow(DEGdf_down2x_FDR5_bf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_bf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_bf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_bf, "../data/deseq2/BG2/BG2_FLUC_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_bf, "../data/deseq2/BG2/BG2_FLUC_DMSO_down.csv")

```

```{r bg2_v_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in BG2 VCZ vs DMSO: %d", nrow(DEGdf_up2x_FDR5_bv))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_bv, n = 10)


sprintf("Downregulated in BG2 VCZ vs DMSO: %d", nrow(DEGdf_down2x_FDR5_bv))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_bv, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_bv, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_bv, "../data/deseq2/BG2/BG2_VORI_DMSO_up.csv")
write.csv(DEGdf_down2x_FDR5_bv, "../data/deseq2/BG2/BG2_VORI_DMSO_down.csv")

```

```{r bg2_vf_deseq_print, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE}

sprintf("Upregulated in BG2 VCZ vs FCZ: %d", nrow(DEGdf_up2x_FDR5_bvf))
# sprintf("Printing the first 10:")
# print(DEGdf_up2x_FDR5_bvf, n = 10)


sprintf("Downregulated in BG2 VCZ vs FCZ: %d", nrow(DEGdf_down2x_FDR5_bvf))
# print("Printing the first 10:")
# print(DEGdf_down2x_FDR5_bvf, n = 10)

# print("Non significant genes: ")
# print(DEGdf_non_significant_bvf, n = 10)

# save files
write.csv(DEGdf_up2x_FDR5_bvf, "../data/deseq2/BG2/BG2_VORI_FLUC_up.csv")
write.csv(DEGdf_down2x_FDR5_bvf, "../data/deseq2/BG2/BG2_VORI_FLUC_down.csv")

```

**Conclusions:**

- no difference between FCZ and VCZ effect in either strain.
- small but comparable number of up/downregulated genes in both strains.
- the CBS138_FLUC outlier impacts the differential expression analysis, but not in a way that changes the overall conclusions, so it is reasonable to proceed with the analysis on all 3 replicates.


### Identification of corresponding genes in S. cerevisiae and between the two C.glabrata strains

Using curated orthologs between C.glabrata CBS138 and S.cerevisiae (downloaded from candidagenome.org), I identified standard and systematic _S. cerevisiae_ names for _C.glabrata_ genes up- and down-regulated in FCZ vs DMSO.
There is no need to do repeat it in VCZ, because there are no significant DEG between these two conditions.

Orthologs are also identified between C.glabrata strains using Reciprocal Blast Best Hits.

```{r ortho_sc, eval=TRUE, echo=FALSE, results="hide", message=FALSE, warning=FALSE}
ortho_cgob = read.table("../data/cgob/C_glabrata_CBS138_S_cerevisiae_orthologs.txt", sep="\t", comment.char = "#", header=FALSE)
colnames(ortho_cgob) = c("systematic_cg", "standard_cg", "database_cg", "systematic_sc", "standard_sc", "database_sc")
head(ortho_cgob)

ortho_up = merge(DEGdf_up2x_FDR5_cf, ortho_cgob, by.x="gene", by.y="systematic_cg")
head(ortho_up)

ortho_down = merge(DEGdf_down2x_FDR5_cf, ortho_cgob, by.x="gene", by.y="systematic_cg")
head(ortho_down)

write.csv(ortho_up, "../data/deseq2/CBS138/CBS138_FLUC_DMSO_up_ortho.csv")
write.csv(ortho_down, "../data/deseq2/CBS138/CBS138_FLUC_DMSO_down_ortho.csv")
```

Here I look at overlap between genes upregulated/downregulated in FCZ treatment between CBS138 and BG2.
This includes:

- reporting the number of genes that are DE in one strain and have a homolog in the other - this shows how many genes unique for the strain are DE
- reporting the number of genes that are DE in both strain - this shows whether the response is similar between CBS138 and BG2.

```{r ortho_cb, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE}

ortho_cb_c_up = merge(DEGdf_up2x_FDR5_cf, rbbh, by.x="gene", by.y="gene_id_cbs138")
# head(ortho_cb_c_up)

ortho_cb_c_down = merge(DEGdf_down2x_FDR5_cf, rbbh, by.x="gene", by.y="gene_id_cbs138") # upregulated in CBS138 with a homolog in BG2
# head(ortho_cb_c_down)

ortho_cb_b_up = merge(DEGdf_up2x_FDR5_bf, rbbh, by.x="gene", by.y="gene_id_bg2")
# head(ortho_cb_b_up)

ortho_cb_b_down = merge(DEGdf_down2x_FDR5_bf, rbbh, by.x="gene", by.y="gene_id_bg2")
# head(ortho_cb_b_down)

sprintf("Number of FCZ upregulated genes in CBS138: %d that have a homolog in BG2: %d", nrow(DEGdf_up2x_FDR5_cf), nrow(ortho_cb_c_up))
sprintf("Number of FCZ downregulated genes in CBS138: %d that have a homolog in BG2: %d", nrow(DEGdf_down2x_FDR5_cf), nrow(ortho_cb_c_down))

sprintf("Number of FCZ upregulated genes in BG2: %d that have a homolog in CBS138: %d", nrow(DEGdf_up2x_FDR5_bf), nrow(ortho_cb_b_up))
sprintf("Number of FCZ downregulated genes in BG2: %d that have a homolog in CBS138: %d", nrow(DEGdf_down2x_FDR5_bf), nrow(ortho_cb_b_down))

ortho_cb_cb_up = merge(ortho_cb_c_up, ortho_cb_b_up, by.x = "gene", by.y = "gene_id_cbs138")
# head(ortho_cb_cb_up)

ortho_cb_cb_down = merge(ortho_cb_c_down, ortho_cb_b_down, by.x = "gene", by.y = "gene_id_cbs138")
# head(ortho_cb_cb_down)

sprintf("Number of FCZ upregulated genes overlapping between CBS138 and BG2: %d", nrow(ortho_cb_cb_up))
sprintf("Number of FCZ downregulated genes overlapping between CBS138 and BG2: %d", nrow(ortho_cb_cb_down))

write.csv(ortho_cb_c_up, "../data/ortho/CBS138_BG2_FLUC_DMSO_CBS138_up_ortho.csv")
write.csv(ortho_cb_c_down, "../data/ortho/CBS138_BG2_FLUC_DMSO_CBS138_down_ortho.csv")
write.csv(ortho_cb_b_up, "../data/ortho/CBS138_BG2_FLUC_DMSO_BG2_up_ortho.csv")
write.csv(ortho_cb_b_down, "../data/ortho/CBS138_BG2_FLUC_DMSO_BG2_down_ortho.csv")

write.csv(ortho_cb_cb_up, "../data/ortho/CBS138_BG2_FLUC_DMSO_common_up_ortho.csv")
write.csv(ortho_cb_cb_down, "../data/ortho/CBS138_BG2_FLUC_DMSO_common_down_ortho.csv")

```

**Conclusions:**

- most upregulated genes have homologs in both strains
- only a little more than 50% of genes differentially expressed in one strain are also differentially expressed in the other


### Gene Ontology analysis of up- and down-regulated genes in CBS138 and BG2

Based on this, I used GO term enrichment (candidagenome.org) of up- and downregulated genes in CBS138, BG2 and common to both strains.

The results are in ../data/go/ folder.

#### Upregulated genes

Both strains up (selected):

- secondary alcohol biosynthetic process
- sterol and lipid biosynthetic process
- sulfur compound metabolic process
- secondary alcohol metabolic process
- small molecule metabolic process
- alcohol biosynthetic process
- small molecule biosynthetic process
- organic hydroxy compound biosynthetic process
- homoserine metabolic process
- non-proteinogenic amino acid metabolic process
- sulfur amino acid biosynthetic process
- aspartate family amino acid metabolic process
- amino acid import across plasma membrane
- L-methionine biosynthetic process

Only CBS138 up:

- carbohydrate metabolic process
- oligosaccharide metabolic process
- monosaccharide metabolic process
- trehalose metabolic process

BG2 only up (selected):

- lysine biosynthetic process via aminoadipic acid
- proteinogenic amino acid metabolic process
- L-amino acid metabolic process
- aspartate family amino acid biosynthetic process


**Conclusions:**

- both strains have upregulation of ergosterol and lipid biosynthesis genes and small molecule metabolism genes (which also includes ERG genes)
- BG2 DEG are more enriched in amino acid metabolism genes
- CBS138 are more enriched in carbohydrate metabolism genes - I think these are generally upregulated in stress?

#### Downregulated genes

Both strains down (selected):

- ribosome biogenes
- nucleic acid catabolism

CBS138 only:

- arginine biosynthetic process
- arginine metabolic process
- glutamine family amino acid biosynthetic process
- ornithine metabolic process
- arginine biosynthetic process via ornithine
- glutamine family amino acid metabolic process
- ribosome biogenesis

BG2 only:

- translation
- ribosome biogenesis

**Conclusions:**

- in both strains mostly ribosome biogenesis genes are downregulated


### Comparison with transcription factor targets

Using data from two publications, both in CBS138:

- UPC2A https://doi.org/10.1371/journal.pgen.1009582
- PDR1 https://doi.org/10.1128/aac.03921-14

I looked at differential expression of two transcription factor targets involved in antifungal resistance.
This part of the analysis in CBS138 only.

This is not enrichment analysis, just search for overlapping genes.

For comparison, I'm also showing targets of _S. cerevisiae_ homologs of these two genes (targets downloaded from SGD).


```{r transcription_factors, echo=FALSE}
pdr1_targets_cg = read.csv("../data/supplementary_material_PDR1/table1.csv")
pdr1_targets_cg_up = merge(ortho_up, pdr1_targets_cg, by.x="gene", by.y="systematic_cg")
pdr1_targets_cg_down = merge(ortho_down, pdr1_targets_cg, by.x="gene", by.y="systematic_cg")
# head(pdr1_targets_cg_up)

sprintf("Number of PDR1 targets in C.glabrata: %d, number of upregulated genes: %d, number of downregulated genes: %d", nrow(pdr1_targets_cg), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
pdr1_targets_cg_up$standard_sc.x
print("Intersection DOWN:")
pdr1_targets_cg_down$standard_sc.x

pdr1_targets = read.csv("../data/transcription_factors/PDR1_targets.txt", sep="\t", comment.char="!")
# head(pdr1_targets)
pdr1_targets_up = merge(ortho_up, pdr1_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")
pdr1_targets_down = merge(ortho_down, pdr1_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")

sprintf("Number of PDR1 targets in S. cerevisiae: %d, , number of upregulated genes: %d, number of downregulated genes: %d", nrow(pdr1_targets), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
pdr1_targets_up$standard_sc
print("Intersection DOWN:")
pdr1_targets_down$standard_sc


upc2a_targets = read.csv("../data/supplementary_material_UPC2A/10_1371_journal_pgen_1009582/table_1b.csv")
upc2a_targets_up = merge(ortho_up, upc2a_targets, by.x="gene", by.y="Gene.ID")
upc2a_targets_down = merge(ortho_down, upc2a_targets, by.x="gene", by.y="Gene.ID")

sprintf("Number of UPC2A targets in C. glabrata: %d, , number of upregulated genes: %d, number of downregulated genes: %d", nrow(upc2a_targets), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
upc2a_targets_up$standard_sc
print("Intersection DOWN:")
upc2a_targets_down$standard_sc

upc2_targets = read.csv("../data/transcription_factors/UPC2_targets_manual.txt", sep="\t", comment.char="!")
# head(upc2_targets)
upc2_targets_up = merge(ortho_up, upc2_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")
upc2_targets_down = merge(ortho_down, upc2_targets, by.x="systematic_sc", by.y="Target.Systematic.Name")

sprintf("Number of UPC2 targets in S. cerevisiae: %d, , number of upregulated genes: %d, number of downregulated genes: %d", nrow(upc2_targets), nrow(ortho_up),  nrow(ortho_down))
print("Intersection UP:")
upc2_targets_up$standard_sc
print("Intersection DOWN:")
upc2_targets_down$standard_sc

```

**Conclusions:**

- I'm not sure how reliable this data is, for example most ERG genes do not show up among _C. glabrata_ UPC2A targets, but do show up among _S. cerevisiae_ Upc2; especially that the UPC2A datasets has more than 10x genes than UPC2.
- given that there is little overlap between CgPdr1 and ScPdr1 targets, I don't know if it's worth looking at S. cerevisiae orthologs
- using binding motifs might be a better idea.

## Differential gene expression comparison between the two strains


```{r c_b_d, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

dds_compared_conditions = 
  run_deseq_vsstrain(counts_all_joined,
                     conditions_to_compare = c("CBS138_RPMI_DMSO", "BG2_RPMI_DMSO"))

dds_compared_conditions

results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)

deseq_df_cbd <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])
write.csv(deseq_df_cbd, "../data/deseq2/CompareStrains/CBS138_BG2_DMSO_allDESeq.csv")

DEGdf_up2x_FDR5_cbd = get_upregulated_genes(deseq_df_cbd)
DEGdf_down2x_FDR5_cbd = get_downregulated_genes(deseq_df_cbd)
DEGdf_non_significant_cbd = get_non_significant_genes(deseq_df_cbd)

print(DEGdf_up2x_FDR5_cbd, n = 20)

print(DEGdf_down2x_FDR5_cbd, n = 20)

print(DEGdf_non_significant_cbd, n = 20)

p_cbd = volcano_plot(deseq_df_cbd, DEGdf_up2x_FDR5_cbd, DEGdf_down2x_FDR5_cbd,
                    xlabel = "log2 fold-change\n← up in BG2       up in CBS138 →") +
  ggtitle("Strain comparison in RPMI DMSO")
#p_cbd

print("Upregulated genes:")
nrow(DEGdf_up2x_FDR5_cbd)
print("Downregulated genes")
nrow(DEGdf_down2x_FDR5_cbd)
```

```{r c_b_v, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

dds_compared_conditions = 
  run_deseq_vsstrain(counts_all_joined,
                     conditions_to_compare = c("CBS138_RPMI_VORI", "BG2_RPMI_VORI"))

dds_compared_conditions

results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)

deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])
write.csv(deseq_df, "../data/deseq2/CompareStrains/CBS138_BG2_VORI_allDESeq.csv")

DEGdf_up2x_FDR5_cbv = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cbv = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cbv = get_non_significant_genes(deseq_df)

print(DEGdf_up2x_FDR5_cbv, n = 20)

print(DEGdf_down2x_FDR5_cbv, n = 20)

print(DEGdf_non_significant_cbv, n = 20)

p_cbv = volcano_plot(deseq_df, DEGdf_up2x_FDR5_cbv, DEGdf_down2x_FDR5_cbv,
                  xlabel = "log2 fold-change\n← up in BG2       up in CBS138 →") +
  ggtitle("Strain comparison in RPMI VCZ")
#p_cbv

print("Upregulated genes:")
nrow(DEGdf_up2x_FDR5_cbv)
print("Downregulated genes")
nrow(DEGdf_down2x_FDR5_cbv)
```

```{r c_b_f, fig.height=6, fig.width=8, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

dds_compared_conditions = 
  run_deseq_vsstrain(counts_all_joined,
                     conditions_to_compare = c("CBS138_RPMI_FLUC", "BG2_RPMI_FLUC"))
dds_compared_conditions

results(dds_compared_conditions)
rN = resultsNames(dds_compared_conditions)
rN

res <- results(dds_compared_conditions, name=rN[2])
head(res)

deseq_df <- deseq_df_transform(dds_compared_conditions, 
                               termstokeep = rN[2])
write.csv(deseq_df, "../data/deseq2/CompareStrains/CBS138_BG2_FLUC_allDESeq.csv")

DEGdf_up2x_FDR5_cbf = get_upregulated_genes(deseq_df)
DEGdf_down2x_FDR5_cbf = get_downregulated_genes(deseq_df)
DEGdf_non_significant_cbf = get_non_significant_genes(deseq_df)

print(DEGdf_up2x_FDR5_cbf, n = 20)

print(DEGdf_down2x_FDR5_cbf, n = 20)

print(DEGdf_non_significant_cbf, n = 20)

p_cbf = volcano_plot(deseq_df, DEGdf_up2x_FDR5_cbf, DEGdf_down2x_FDR5_cbf,
                    xlabel = "log2 fold-change\n← up in BG2       up in CBS138 →") +
  ggtitle("Strain comparison in RPMI FCZ")
#p_cbf

print("Upregulated genes:")
nrow(DEGdf_up2x_FDR5_cbf)
print("Downregulated genes")
nrow(DEGdf_down2x_FDR5_cbf)
```

```{r arrange_strain_volcano_plots, fig.height=7.5, fig.width=3.5, echo=FALSE, message=FALSE, warning=FALSE}
pg = plot_grid(p_cbd, p_cbv, p_cbf, ncol = 1)
pg
```

Report number of differentially expressed genes in CBS138 vs BG2:

```{r diff_print, echo=FALSE}

sprintf("DMSO upregulated: %d, downregulated: %d", nrow(DEGdf_up2x_FDR5_cbd), nrow(DEGdf_down2x_FDR5_cbd))

sprintf("FCZ upregulated: %d, downregulated: %d", nrow(DEGdf_up2x_FDR5_cbf), nrow(DEGdf_down2x_FDR5_cbf))

sprintf("VCZ upregulated: %d, downregulated: %d", nrow(DEGdf_up2x_FDR5_cbv), nrow(DEGdf_down2x_FDR5_cbv))

write.csv(DEGdf_up2x_FDR5_cbd, "../data/deseq2/CompareStrains/CBS138_BG2_DMSO_up.csv")

write.csv(DEGdf_down2x_FDR5_cbd, "../data/deseq2/CompareStrains/CBS138_BG2_DMSO_down.csv")

write.csv(DEGdf_up2x_FDR5_cbf, "../data/deseq2/CompareStrains/CBS138_BG2_FLUC_up.csv")

write.csv(DEGdf_down2x_FDR5_cbf, "../data/deseq2/CompareStrains/CBS138_BG2_FLUC_down.csv")

write.csv(DEGdf_up2x_FDR5_cbv, "../data/deseq2/CompareStrains/CBS138_BG2_VORI_up.csv")

write.csv(DEGdf_down2x_FDR5_cbv, "../data/deseq2/CompareStrains/CBS138_BG2_VORI_down.csv")


```

### GO analysis of strain differences

DMSO upregulated in CBS138 (selected):

- amino acid biosynthesis
- cell adhesion
- spore wall assembly
- cell wall assembly

DMSO downregulated in CBS138 (selected):

- trehalose metabolism
- small molecule catabolism
- lipid catabolism

FCZ upregulated in CBS138:

- cell adhesion: AWP2, CAGL0A01584g, CAGL0I09856g, CAGL0K12078g, EPA23, EPA6

FCZ downregulated in CBS138 (selected):

- small molecule catabolism
- energy derivation by oxidation of organic compounds

VCZ upregulated in CBS138:

- cell adhesion: AWP2, CAGL0A01584g, CAGL0I09856g, EPA23, EPA3, EPA6
- filamentous growth: CAGL0H02145g, CAGL0I01254g, CAGL0I09856g, CAGL0K04169g, CAGL0M03663g, CAGL0M09042g, EFG1, SHO1, TEC1, YPS2

VCZ downregulated in CBS138:

- small molecule catabolism
- trehalose metabolism
- carbohydrate metabolism
- energy derivation by oxidation of organic compounds


## Analysis of Drug vs no Drug and strain differences

The above conclusions for pairwise comparisons suggest that we should compare:

- Drug vs no drug, taking average of VCZ and FCZ responses
- and compare across strains to get the strain-specific drug interactions

### Differential expression shared across both strains, both drugs

Here, "up" means that both drugs induces more transcript abundance in both strains. Essentially, treating both drugs as biological replicates, and both strains as biological replicates.

```{r drug_bothstrains, fig.height=4, fig.width=5}

sample_sheet_all_drugnodrug <-
  sample_sheet_all %>%
  # select only relevant columns
  dplyr::select(SampleID, Code, Strain, Medium, Rep) %>%
  # make Strain a factor in predictable order 
  mutate(Strain = factor(Strain, 
                         levels = c("BG2", "CBS138"))
         ) %>%
  # create two new columns for Drug, and Voriconazole vs Fluconazole
  mutate(Drug = Medium %in% c("RPMI_FLUC", "RPMI_VORI"),
         VORI = Medium %in% c("RPMI_VORI")) %>%
  magrittr::set_rownames(.$Code)

dds_drugnodrug_bc <- 
  DESeqDataSetFromMatrix(
    countData = counts_all_joined %>%
      dplyr::select(sample_sheet_all_drugnodrug$Code) %>%
      magrittr::set_rownames(counts_all_joined$Gene),
    colData = sample_sheet_all_drugnodrug,
    design = ~ Strain + Drug) %>%
  DESeq()

resultsNames(dds_drugnodrug_bc)

deseq_df_drugnodrug_bc  <- 
  deseq_df_transform(dds_drugnodrug_bc,
                     termstokeep = "DrugTRUE")

deseq_df_drugnodrug_bc

DEGdf_up2x_FDR5_drug = get_upregulated_genes(deseq_df_drugnodrug_bc)
DEGdf_down2x_FDR5_drug = get_downregulated_genes(deseq_df_drugnodrug_bc)

print(DEGdf_up2x_FDR5_drug, n = 20)

print(DEGdf_down2x_FDR5_drug, n = 20)

p_drugnodrug_bc = 
  volcano_plot(deseq_df_drugnodrug_bc, 
               DEGdf_up2x_FDR5_drug, 
               DEGdf_down2x_FDR5_drug,
               xlabel = "log2 fold-change\n← down in drug       up in drug →") +
  ggtitle("Azole vs DMSO, in both strains")
p_drugnodrug_bc

write.csv(deseq_df_drugnodrug_bc, "../data/deseq2/DrugBothStrains/BothStrains_Drug_vs_DMSO_allDESeq.csv")
write.csv(DEGdf_up2x_FDR5_drug, "../data/deseq2/DrugBothStrains/BothStrains_Drug_vs_DMSO_up2x_FDR5.csv")
writeLines(DEGdf_up2x_FDR5_drug$gene, "../data/deseq2/DrugBothStrains/BothStrains_Drug_vs_DMSO_up2x_FDR5_names.txt")
write.csv(DEGdf_down2x_FDR5_drug, "../data/deseq2/DrugBothStrains/BothStrains_Drug_vs_DMSO_down2x_FDR5.csv")
writeLines(DEGdf_down2x_FDR5_drug$gene, "../data/deseq2/DrugBothStrains/BothStrains_Drug_vs_DMSO_down2x_FDR5_names.txt")

```

### Differential drug-strain interactions across strains

We're looking for differential strain responses to drug. Here, "up" means that azoles induce more change in transcript abundance in CBS138 as compared to BG2. Again, treating both drugs as biological replicates, due to the small differences in response to the two diferent drugs.

```{r drug_strain_interactions, fig.height=4, fig.width=5}

dds_drugstrain <- 
  DESeqDataSetFromMatrix(
    countData = counts_all_joined %>%
      dplyr::select(sample_sheet_all_drugnodrug$Code) %>%
      magrittr::set_rownames(counts_all_joined$Gene),
    colData = sample_sheet_all_drugnodrug,
    design = ~ Strain * Drug) %>%
  DESeq()

resultsNames(dds_drugstrain)

deseq_df_drugstrain  <- 
  deseq_df_transform(dds_drugstrain,
                     termstokeep = "StrainCBS138.DrugTRUE")

deseq_df_drugstrain

DEGdf_up2x_FDR5_drugstrain = get_upregulated_genes(deseq_df_drugstrain)
DEGdf_down2x_FDR5_drugstrain = get_downregulated_genes(deseq_df_drugstrain)

print(DEGdf_up2x_FDR5_drugstrain, n = 20)

print(DEGdf_down2x_FDR5_drugstrain, n = 20)

p_drugstrain = volcano_plot(deseq_df_drugstrain, DEGdf_up2x_FDR5_drugstrain, DEGdf_down2x_FDR5_drugstrain,
             ylimits = c(0,5),
               xlabel = "log2 fold-change\n← more change in BG2       more change in CBS138 →") +
  ggtitle("Azole vs DMSO, strain differences")
p_drugstrain

write.csv(deseq_df_drugstrain, "../data/deseq2/DrugStrainInteractions/DrugStrain_CBS138_vs_BG2_allDESeq.csv")
write.csv(DEGdf_up2x_FDR5_drugstrain, "../data/deseq2/DrugStrainInteractions/DrugStrain_CBS138_vs_BG2_up2x_FDR5.csv")
writeLines(DEGdf_up2x_FDR5_drugstrain$gene, "../data/deseq2/DrugStrainInteractions/DrugStrain_CBS138_vs_BG2_up2x_FDR5_names.txt")
write.csv(DEGdf_down2x_FDR5_drugstrain, "../data/deseq2/DrugStrainInteractions/DrugStrain_CBS138_vs_BG2_down2x_FDR5.csv")
writeLines(DEGdf_down2x_FDR5_drugstrain$gene, "../data/deseq2/DrugStrainInteractions/DrugStrain_CBS138_vs_BG2_down2x_FDR5_names.txt")

```


### Volcano plot drug-strain interactions with highlighted genes

Highlighting calcium transporters MID1, ECM7, CCH1.

```{r drug_strain_interactions_genenames, fig.height=7.5, fig.width = 4}
# Warning! This code chunk is based on manually reading in a file with gene names only of calculated deseq genes. It would be better to do a merge on all Cg gene names.

calcium_transporters <-
  tribble(~gene, ~name,
          "CAGL0B02211g", "CCH1",
          "CAGL0M03597g", "MID1",
          "CAGL0M00748g", "ECM7")

favourite_TFs <- 
  tribble(~gene, ~name,
          "CAGL0M08800g", "YAP6",
          "CAGL0L06776g", "GAT3/4")

drugstrain_down_names <- 
  read_tsv("../data/deseq2/DrugStrainInteractions/DrugStrain_CBS138_vs_BG2_down2x_FDR5_fungidbtable.txt") %>%
  select(gene, name) %>%
  filter(!is.na(name))

p_summary_volcanos_labels =
  plot_grid(
    p_cbd + 
      geom_label_repel(data = deseq_df_cbd %>%
                         right_join(calcium_transporters, by = "gene"),
                       aes(label = name), 
                       min.segment.length = 0, max.overlaps = 20,
                       box.padding = 0.1, label.padding = 0.1),
    p_drugnodrug_bc + 
      geom_label_repel(data = deseq_df_drugnodrug_bc %>%
                         right_join(calcium_transporters, by = "gene"),
                       aes(label = name), 
                       min.segment.length = 0, max.overlaps = 20,
                       box.padding = 0.1, label.padding = 0.1),
    p_drugstrain + 
      geom_label_repel(data = deseq_df_drugstrain %>%
                         right_join(
                           bind_rows(calcium_transporters, 
                                    favourite_TFs,
                                    drugstrain_down_names),
                           by = "gene"),
                       aes(label = name), 
                       min.segment.length = 0, max.overlaps = 20,
                       box.padding = 0.1, label.padding = 0.1),
    ncol = 1, nrow = 3)

p_summary_volcanos_labels

ggsave("../figures/summary_volcano_plots_genelabels.png", 
       plot = p_summary_volcanos_labels,
       width = 7.5, height = 9)
```


## Venn Diagram of DEG lists

```{r venn}
library(VennDiagram)
 
# Generate 3 sets of 200 words
set1 <- paste(rep("word_" , 200) , sample(c(1:1000) , 200 , replace=F) , sep="")
set2 <- paste(rep("word_" , 200) , sample(c(1:1000) , 200 , replace=F) , sep="")
set3 <- paste(rep("word_" , 200) , sample(c(1:1000) , 200 , replace=F) , sep="")
 
set.seed(20190708)
genes <- paste("gene",1:1000,sep="")
x <- list(
  A = sample(genes,300), 
  B = sample(genes,525), 
  C = sample(genes,440),
  D = sample(genes,350)
  )

# Chart
venn.diagram(
  x = list(set1, set2, set3),
  category.names = c("Set 1" , "Set 2 " , "Set 3"),
  filename = NULL,
  output=TRUE
)

display_venn <- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}
display_venn(
  x,
  category.names = c("Set 1" , "Set 2 " , "Set 3", "Set 4"),
  fill = c("#999999", "#E69F00", "#56B4E9", "#009E73")
  )
display_venn(x)

library("ggVennDiagram")
ggVennDiagram(x, label_alpha = 0)

cf_up = as.vector(ortho_cb_c_up$gene)
cbd_up = as.vector(DEGdf_up2x_FDR5_cbd$gene)
cbf_up = as.vector(DEGdf_up2x_FDR5_cbf$gene)
bf_up = as.vector(ortho_cb_b_up$gene_id_cbs138)

y <- list(
  CBS138_FLUC = cf_up, 
  strain_DMSO = cbd_up, 
  strain_FLUC = cbf_up,
  BG2_FLUC = bf_up
  )

display_venn(
  y,
  fill = c("#999999", "#E69F00", "#56B4E9", "#009E73")
  )


print(head(DEGdf_up2x_FDR5_cbd))
```


```{python interpretation_figures_rlog, fig.height = 2, fig.width=2, eval=FALSE}
import pandas as pd

plt.rcParams['svg.fonttype'] = 'none'

rlog_all_p = r.rlog_all_df
rlog_all_p = rlog_all_p.reset_index().rename(columns={'index': 'Geneid'})


filamentous_genes = ["CAGL0H02145g",	"CAGL0I01254g",	"CAGL0I09856g", "CAGL0K04169g", "CAGL0M03663g", "CAGL0M09042g"]
cell_adhesion_genes = ["CAGL0K00110g", "CAGL0A01584g", "CAGL0I09856g", "CAGL0K12078g", "CAGL0I00220g", "CAGL0E06688g", "CAGL0C00110g"]

s = rlog_all_p[rlog_all_p["Geneid"].isin(cell_adhesion_genes)]

t = s.set_index("Geneid")
#plt.figure(figsize=(10, 10))
#plt.subplots_adjust(left=0.5, right=1.5, bottom=4.0)
g = sns.clustermap(t, cmap='vlag', method="single", annot_kws={"size": 7})
plt.setp(g.ax_heatmap.get_yticklabels(), rotation=0)  # For y axis
plt.show()
#plt.savefig("/Users/s1964600/Work/random_figures/new_svg/cpac_scer_deg.svg")
plt.close()

```


# Composite figures

Main manuscript RNA-seq composite plot.

```{r summary_plot, fig.height=7.5, fig.width = 7, echo=FALSE, message=FALSE, warning=FALSE}
guide_legend_pc12 = 
  guide_legend("Strain,\nTreatment",
               position = "bottom",
               nrow = 3, ncol = 2)
plot_margin_summary_plot = 
              theme(plot.margin = 
                      unit(c(12, 12, 12, 12), 
                                "points")) 

p_summary_leftcolumn =
  plot_grid(NULL, 
            pc12 + 
              plot_margin_summary_plot + 
              guides(colour = guide_legend_pc12, 
                     shape = guide_legend_pc12), 
            plot_percentvar_rlog +
              theme(plot.margin = 
                      unit(c(6, 12, 6, 30), 
                                "points")) , 
            rel_heights = c(2,3,1),
            ncol = 1,
            labels = c("A","B","C")) +
  draw_text("EXPERIMENTAL DESIGN HERE", x = 0.5, y = 0.9)
  

p_summary_volcanos =
  plot_grid(p_cbd + plot_margin_summary_plot, 
            p_drugnodrug_bc + plot_margin_summary_plot, 
            p_drugstrain + plot_margin_summary_plot, 
            ncol = 1, nrow = 3,
            align = "v",
            labels = c("D","E","F"))

p_summary = 
  plot_grid(p_summary_leftcolumn, 
            p_summary_volcanos, 
            ncol = 2)
p_summary

ggsave("../figures/RNAseqsummmary_nodesign.eps",
       p_summary, 
       height=7.5, width = 7)
ggsave("../figures/RNAseqsummmary_nodesign.png",
       p_summary, 
       height=7.5, width = 7)
```

## Load gene info

Load gene info from FungiDB with some of the important gene names.

```{r load_geneinfo}
CBS138_geneinfo <- 
  read_tsv("../data/genomes/CBS138_allgenes.txt",
           col_names = c("Geneid", "Transcriptid","GenomicLocation", "Description", "Name"),
           skip = 1,
           na = c("", "NA", "N/A"))

BG2_geneinfo <- 
  read_tsv("../data/genomes/BG2_allgenes.txt",
           col_names = c("Geneid", "Transcriptid","GenomicLocation", "Description"),
           skip = 1,
           na = c("", "NA", "N/A")) %>%
  mutate(Name = stringr::str_extract(Description, "^[A-Z]+[0-9]+"))
```

## Plot groups of genes

```{r plot_fewgenes, echo = FALSE}

getrpms <- function(namelist,
                    geneinfodf = CBS138_geneinfo,
                    rpmdf = counts_rpm_CBS138,
                    namelist_order = TRUE) {
  
  mygenerpmdf <- 
    geneinfodf %>%
    dplyr::select(Geneid, Name) %>%
    dplyr::filter(Name %in% namelist) %>%
    left_join(rpmdf, by = "Geneid") 
  
  if(namelist_order) {
    mygenerpmdf <- mutate(mygenerpmdf, 
                           Name = factor(Name, levels = namelist))
  }
  
  mygenerpmdf
}

getrpmslong <- function(namelist,
                        geneinfodf = CBS138_geneinfo,
                        rpmdf = counts_rpm_CBS138,
                        namelist_order = TRUE) {
  
  getrpms(namelist = namelist,
          geneinfodf = geneinfodf,
          rpmdf = rpmdf,
          namelist_order = namelist_order) %>% 
    pivot_longer(cols = -c(Geneid, Name),
                 names_to = "Sample", values_to = "RPM") %>%
    separate(Sample, 
             into = c("Strain", "Media", "Treatment", "BioRep"),
             remove = FALSE)
}

plot_fewgenes_rpmpointsummary <- 
  function(namelist) {
    rpmdf_long <- bind_rows(
      getrpmslong(namelist, CBS138_geneinfo, counts_rpm_CBS138),
      getrpmslong(namelist, BG2_geneinfo,    counts_rpm_BG2)
    )
    
    ggplot(data = rpmdf_long,
           aes(x = Name, y = RPM, colour = Treatment, shape = Strain)) +
      geom_point(position = 
                   position_jitter(height = 0, 
                                   width = 0.15, 
                                   seed = 1),
                 size = 1) + 
      scale_y_log10() +
      stat_summary(fun.min="mean",fun.max="mean",
                   geom="errorbar", linewidth = 1, width = 0.6,
                   aes(group=Treatment)) +
      scale_colour_manual(
        values = c("DMSO" = "grey20",
                   "FLUC" = "purple3",
                   "VORI" = "green4"),
        labels = c("DMSO" = "DMSO",
                   "FLUC" = "FCZ",
                   "VORI" = "VCZ")) +
      scale_shape_manual(values = c("CBS138" = 16,
                                    "BG2" = 17)) +
      coord_flip() +
      facet_wrap(~ Strain, ncol = 2) +
      theme(panel.border = element_rect(colour = "grey50"),
            panel.grid.major.x = element_line(colour = "grey90"))
  }

plot_fewgenes_rpmheatmap <- 
  function(namelist) {
    rpmdf_long <- bind_rows(
      getrpmslong(namelist, CBS138_geneinfo, counts_rpm_CBS138),
      getrpmslong(namelist, BG2_geneinfo,    counts_rpm_BG2)
    )
    
    ggplot(data = rpmdf_long,
           aes(y = Name, x = Sample, fill = log10(RPM))) +
      geom_tile() +
      theme(axis.text.x = element_text(angle = 90)) +
      scale_fill_distiller(palette = "Spectral")
  }
```

```{r plot_rpmpointsummaryYPSetc, fig.height = 4, fig.width = 6, echo = FALSE, message = FALSE}
plot_fewgenes_rpmpointsummary(paste0("YPS",seq(1,11)))

# plot_fewgenes_rpmpointsummary(paste0("AWP",seq(1,13)))
# plot_fewgenes_rpmpointsummary(paste0("EPA",seq(1,23)))
# plot_fewgenes_rpmpointsummary(paste0("PWP",seq(1,7)))

plot_fewgenes_rpmpointsummary(c("AWP1", "AWP2", "EPA1", "EPA2", "EPA3", "EPA6", "EPA23"))

plot_fewgenes_rpmpointsummary(paste0("ERG",seq(1,27)))


plot_fewgenes_rpmpointsummary(c("CCH1", "MID1", "ECM7", "YVC1"))
```


```{r plot_rpmheatmapYPSetc, fig.height = 4, fig.width = 5, echo = FALSE, message = FALSE}
plot_fewgenes_rpmheatmap(paste0("YPS",seq(1,11)))

plot_fewgenes_rpmheatmap(c("AWP1", "AWP2", "EPA1", "EPA2", "EPA3", "EPA6", "EPA23"))

plot_fewgenes_rpmheatmap(paste0("ERG",seq(1,27)))

```



# TO DOs:

- [ ] look at the orthogroups for differences between the species
- [ ] change data formatting to have one file with orthology information instead of multiple files for every condition
- [x] extend the analysis of regulation, including the strain differences
- [ ] less text, more figures
- [ ] add calcium-related gene names to figures
